diff --git a/lib/index.js b/lib/index.js
index c991f62dc64553502e9911a7f21e77e008d7f438..e503c7494d21b13df85b10e1657b2af8ca4d964f 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -222,7 +222,6 @@ var _transform = require("./transform");
 var _transformFile = require("./transform-file");
 var _transformAst = require("./transform-ast");
 var _parse = require("./parse");
-var thisFile = require("./index");
 const version = "7.21.5";
 exports.version = version;
 const DEFAULT_EXTENSIONS = Object.freeze([".js", ".jsx", ".es6", ".es", ".mjs", ".cjs"]);
