diff --git a/node_modules/@lavamoat/lavapack/src/runtime.js b/node_modules/@lavamoat/lavapack/src/runtime.js
index 04b28f7..d4c89f8 100644
--- a/node_modules/@lavamoat/lavapack/src/runtime.js
+++ b/node_modules/@lavamoat/lavapack/src/runtime.js
@@ -8516,9 +8516,9 @@ optGetStackString = undefined) =>
       // eslint-disable-next-line @endo/no-polymorphic-call
       event.preventDefault();
       // eslint-disable-next-line @endo/no-polymorphic-call
-      const stackString = loggedErrorHandler.getStackString(event.error);
+      // const stackString = loggedErrorHandler.getStackString(event.error);
       // eslint-disable-next-line @endo/no-polymorphic-call
-      causalConsole.error(stackString);
+      causalConsole.error(event.error);
       if (errorTrapping === 'exit' || errorTrapping === 'abort') {
         globalThis.window.location.href = `about:blank`;
       }
