diff --git a/dist/utils/transaction.mjs b/dist/utils/transaction.mjs
index 02b3b37bf2796a9c54f79224c675cd01dfe2831f..12b1ed6f99ef3983cd63316d4724951f66c05057 100644
--- a/dist/utils/transaction.mjs
+++ b/dist/utils/transaction.mjs
@@ -225,7 +225,11 @@ export const getAddTransactionBatchParams = async ({ messenger, isBridgeTx, appr
     // Enable 7702 batching when the quote includes gasless 7702 support,
     // or when the account is already delegated (to avoid the in-flight
     // transaction limit for delegated accounts)
-    const disable7702 = !skipGasFields && !isDelegatedAccount;
+    let disable7702 = !skipGasFields && !isDelegatedAccount;
+    // For gasless transactions with STX/sendBundle we keep disabling 7702.
+    if (gasIncluded && !gasIncluded7702) {
+        disable7702 = true;
+    }
     const transactions = [];
     if (resetApproval) {
         const gasFees = await calculateGasFees(skipGasFields, messenger, estimateGasFeeFn, resetApproval, networkClientId, hexChainId, isGasless ? txFee : undefined);
