diff --git a/dist/window/WindowPostMessageStream.cjs b/dist/window/WindowPostMessageStream.cjs
index 0aaba03f189a09914f71b639d352972fc3f49761..578b2c8237cc78d96df291767dcce1ac5a5fa37e 100644
--- a/dist/window/WindowPostMessageStream.cjs
+++ b/dist/window/WindowPostMessageStream.cjs
@@ -65,7 +65,6 @@ class WindowPostMessageStream extends BasePostMessageStream_1.BasePostMessageStr
         /* eslint-disable @typescript-eslint/no-non-null-assertion */
         if ((this._targetOrigin !== '*' &&
             getOrigin.call(event) !== this._targetOrigin) ||
-            getSource.call(event) !== this._targetWindow ||
             !(0, utils_2.isValidStreamMessage)(message) ||
             message.target !== this._name) {
             return;
diff --git a/dist/window/WindowPostMessageStream.mjs b/dist/window/WindowPostMessageStream.mjs
index db60ec43efa2504b6590165e3162533a4bf1e4cd..2f3110b58eca1556251faf742e2c69cecf020f6e 100644
--- a/dist/window/WindowPostMessageStream.mjs
+++ b/dist/window/WindowPostMessageStream.mjs
@@ -62,7 +62,6 @@ export class WindowPostMessageStream extends BasePostMessageStream {
         /* eslint-disable @typescript-eslint/no-non-null-assertion */
         if ((this._targetOrigin !== '*' &&
             getOrigin.call(event) !== this._targetOrigin) ||
-            getSource.call(event) !== this._targetWindow ||
             !isValidStreamMessage(message) ||
             message.target !== this._name) {
             return;
