diff --git a/dist/TransactionController.js b/dist/TransactionController.js
index 2684c0ac3904f3d56aadc6aee6d4af069178b402..1a49782bf68ed021b39edb184eecfecfed4dc988 100644
--- a/dist/TransactionController.js
+++ b/dist/TransactionController.js
@@ -898,6 +898,9 @@ class TransactionController extends base_controller_1.BaseControllerV1 {
             updatedTransactionMeta.error = (0, utils_1.normalizeTxError)(new Error(errorMessage));
         }
         this.updateTransaction(updatedTransactionMeta, `TransactionController:updateCustodialTransaction - Custodial transaction updated`);
+        if ([types_1.TransactionStatus.submitted, types_1.TransactionStatus.failed].includes(status)) {
+            this.hub.emit(`${transactionMeta.id}:finished`, updatedTransactionMeta);
+        }
     }
     /**
      * Creates approvals for all unapproved transactions persisted.
