diff --git a/dist/bridge-status-controller.cjs b/dist/bridge-status-controller.cjs
index c8dcd1f18fd4395b13ac7945e533ae1a92553950..269c474990ff3f938bd613d239809ccd7c2f1fd1 100644
--- a/dist/bridge-status-controller.cjs
+++ b/dist/bridge-status-controller.cjs
@@ -867,12 +867,12 @@
                     const bridgeTxMetaForHistory = {
                         ...syntheticMeta,
                         id: orderUid,
-                        originalTransactionId: syntheticMeta.id, // Keep original txId for TransactionController updates
                     };
                     // Use orderId as the history key for intent transactions
                     const historyKey = __classPrivateFieldGet(this, _BridgeStatusController_addTxToHistory, "f").call(this, {
                         accountAddress,
                         bridgeTxMeta: bridgeTxMetaForHistory,
+                        originalTransactionId: syntheticMeta.id, // Keep original txId for TransactionController updates
                         quoteResponse,
                         slippagePercentage: 0,
                         isStxEnabled: false,
diff --git a/dist/bridge-status-controller.mjs b/dist/bridge-status-controller.mjs
index 3c285a68a35e69b6898f5c811a000324044213c8..eaf41b8adeed21aeffbdf95d03fdd9eff7910dd1 100644
--- a/dist/bridge-status-controller.mjs
+++ b/dist/bridge-status-controller.mjs
@@ -864,12 +864,12 @@
                     const bridgeTxMetaForHistory = {
                         ...syntheticMeta,
                         id: orderUid,
-                        originalTransactionId: syntheticMeta.id, // Keep original txId for TransactionController updates
                     };
                     // Use orderId as the history key for intent transactions
                     const historyKey = __classPrivateFieldGet(this, _BridgeStatusController_addTxToHistory, "f").call(this, {
                         accountAddress,
                         bridgeTxMeta: bridgeTxMetaForHistory,
+                        originalTransactionId: syntheticMeta.id, // Keep original txId for TransactionController updates
                         quoteResponse,
                         slippagePercentage: 0,
                         isStxEnabled: false,
