diff --git a/dist/chunk-4ZD3DTQ7.js b/dist/chunk-4ZD3DTQ7.js
index e172d15715b3cb4f26c42e51f8c7ff7394075bfe..2c32427d0e005c8db2e4aa524609972fd57cba60 100644
--- a/dist/chunk-4ZD3DTQ7.js
+++ b/dist/chunk-4ZD3DTQ7.js
@@ -341,7 +341,6 @@ var NetworkController = class extends _basecontroller.BaseController {
   async initializeProvider() {
     _chunkZ4BLTVTBjs.__privateMethod.call(void 0, this, _ensureAutoManagedNetworkClientRegistryPopulated, ensureAutoManagedNetworkClientRegistryPopulated_fn).call(this);
     _chunkZ4BLTVTBjs.__privateMethod.call(void 0, this, _applyNetworkSelection, applyNetworkSelection_fn).call(this);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
diff --git a/dist/chunk-UG2NYGJD.mjs b/dist/chunk-UG2NYGJD.mjs
index c39eb49a4a1d2b4ddb78aadb4fb03446b1705528..73f4df206f489036ed52831ac685219074bb768b 100644
--- a/dist/chunk-UG2NYGJD.mjs
+++ b/dist/chunk-UG2NYGJD.mjs
@@ -341,7 +341,6 @@ var NetworkController = class extends BaseController {
   async initializeProvider() {
     __privateMethod(this, _ensureAutoManagedNetworkClientRegistryPopulated, ensureAutoManagedNetworkClientRegistryPopulated_fn).call(this);
     __privateMethod(this, _applyNetworkSelection, applyNetworkSelection_fn).call(this);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
