diff --git a/dist/permitted/getClientStatus.cjs b/dist/permitted/getClientStatus.cjs
index aac07b4bf23798e5e6fa2ea7deca324d7563df1b..5448bd4c5111ba82e65a5a9fef36866a007bfe90 100644
--- a/dist/permitted/getClientStatus.cjs
+++ b/dist/permitted/getClientStatus.cjs
@@ -3,6 +3,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.getClientStatusHandler = void 0;
 const hookNames = {
     getIsLocked: true,
+    getIsActive: true,
 };
 /**
  * `snap_getClientStatus` returns useful information about the client running the snap.
@@ -23,10 +24,11 @@ exports.getClientStatusHandler = {
  * @param end - The `json-rpc-engine` "end" callback.
  * @param hooks - The RPC method hooks.
  * @param hooks.getIsLocked - A function that returns whether the client is locked or not.
+ * @param hooks.getIsActive - A function that returns whether the client is opened or not.
  * @returns Nothing.
  */
-async function getClientStatusImplementation(_request, response, _next, end, { getIsLocked }) {
-    response.result = { locked: getIsLocked() };
+async function getClientStatusImplementation(_request, response, _next, end, { getIsLocked, getIsActive }) {
+    response.result = { locked: getIsLocked(), active: getIsActive() };
     return end();
 }
 //# sourceMappingURL=getClientStatus.cjs.map
\ No newline at end of file
diff --git a/dist/permitted/getClientStatus.mjs b/dist/permitted/getClientStatus.mjs
index 35e693ad0750f70885a9a90e9d11cad55bf3e511..7ee29914180e20ef08581482db9a4ed52ca2034e 100644
--- a/dist/permitted/getClientStatus.mjs
+++ b/dist/permitted/getClientStatus.mjs
@@ -1,5 +1,6 @@
 const hookNames = {
     getIsLocked: true,
+    getIsActive: true,
 };
 /**
  * `snap_getClientStatus` returns useful information about the client running the snap.
@@ -20,10 +21,11 @@ export const getClientStatusHandler = {
  * @param end - The `json-rpc-engine` "end" callback.
  * @param hooks - The RPC method hooks.
  * @param hooks.getIsLocked - A function that returns whether the client is locked or not.
+ * @param hooks.getIsActive - A function that returns whether the client is opened or not.
  * @returns Nothing.
  */
-async function getClientStatusImplementation(_request, response, _next, end, { getIsLocked }) {
-    response.result = { locked: getIsLocked() };
+async function getClientStatusImplementation(_request, response, _next, end, { getIsLocked, getIsActive }) {
+    response.result = { locked: getIsLocked(), active: getIsActive() };
     return end();
 }
 //# sourceMappingURL=getClientStatus.mjs.map
\ No newline at end of file
