diff --git a/dist/caip-types.cjs b/dist/caip-types.cjs
index d6913b98b12a29704e1fb248b179249f920da6f2..661e7ea294d6cbc9f2461d0c071e6ee4de18c303 100644
--- a/dist/caip-types.cjs
+++ b/dist/caip-types.cjs
@@ -32,6 +32,7 @@ var KnownCaipNamespace;
 (function (KnownCaipNamespace) {
     /** EIP-155 compatible chains. */
     KnownCaipNamespace["Eip155"] = "eip155";
+    KnownCaipNamespace["Wallet"] = "wallet";
 })(KnownCaipNamespace = exports.KnownCaipNamespace || (exports.KnownCaipNamespace = {}));
 /**
  * Check if the given value is a {@link CaipChainId}.
diff --git a/dist/caip-types.d.cts b/dist/caip-types.d.cts
index ba5f0820271d53fa642f616e7495da5ac6069ce4..0eef5e28622f3281d51149c67504d7afeb6b9a14 100644
--- a/dist/caip-types.d.cts
+++ b/dist/caip-types.d.cts
@@ -32,7 +32,8 @@ export type CaipAccountAddress = Infer<typeof CaipAccountAddressStruct>;
 /** Known CAIP namespaces. */
 export declare enum KnownCaipNamespace {
     /** EIP-155 compatible chains. */
-    Eip155 = "eip155"
+    Eip155 = "eip155",
+    Wallet = 'wallet',
 }
 /**
  * Check if the given value is a {@link CaipChainId}.
diff --git a/dist/caip-types.d.mts b/dist/caip-types.d.mts
index 1e990705efd42abd04a304e3e4f5cc84e947274f..0960008e1fdc34ebe54c8b23e5a1dcaf6a5dd0e5 100644
--- a/dist/caip-types.d.mts
+++ b/dist/caip-types.d.mts
@@ -32,7 +32,8 @@ export type CaipAccountAddress = Infer<typeof CaipAccountAddressStruct>;
 /** Known CAIP namespaces. */
 export declare enum KnownCaipNamespace {
     /** EIP-155 compatible chains. */
-    Eip155 = "eip155"
+    Eip155 = "eip155",
+    Wallet = 'wallet',
 }
 /**
  * Check if the given value is a {@link CaipChainId}.
diff --git a/dist/caip-types.mjs b/dist/caip-types.mjs
index 5f6c93af75c260ca26f754c573a0780843eea3d9..f24aa936918a15f2ee92bca6d94bfa6f774f2c4e 100644
--- a/dist/caip-types.mjs
+++ b/dist/caip-types.mjs
@@ -29,6 +29,7 @@ export var KnownCaipNamespace;
 (function (KnownCaipNamespace) {
     /** EIP-155 compatible chains. */
     KnownCaipNamespace["Eip155"] = "eip155";
+    KnownCaipNamespace["Wallet"] = "wallet";
 })(KnownCaipNamespace = KnownCaipNamespace || (KnownCaipNamespace = {}));
 /**
  * Check if the given value is a {@link CaipChainId}.
