diff --git a/dist/helpers/IncomingTransactionHelper.cjs b/dist/helpers/IncomingTransactionHelper.cjs
index cd4a8158be0960eaf9d74cc6328e0b3bbd06740d..21336f5b778c05f8cd0469ff9964c704c96f26b1 100644
--- a/dist/helpers/IncomingTransactionHelper.cjs
+++ b/dist/helpers/IncomingTransactionHelper.cjs
@@ -20,7 +20,7 @@ exports.IncomingTransactionHelper = void 0;
 // eslint-disable-next-line import-x/no-nodejs-modules
 const events_1 = __importDefault(require("events"));
 const logger_1 = require("../logger.cjs");
-const INTERVAL = 1000 * 30; // 30 Seconds
+const INTERVAL = 1000 * 60 * 4; // 4 Minutes
 class IncomingTransactionHelper {
     constructor({ getCache, getCurrentAccount, getLocalTransactions, includeTokenTransfers, isEnabled, queryEntireHistory, remoteTransactionSource, trimTransactions, updateCache, updateTransactions, }) {
         _IncomingTransactionHelper_instances.add(this);
diff --git a/dist/helpers/IncomingTransactionHelper.mjs b/dist/helpers/IncomingTransactionHelper.mjs
index 45129a2bdc8f5bcb3955eb7c1a82a1164364a084..138efb1d178be8d2c5dc623ecd18b4997fb49b37 100644
--- a/dist/helpers/IncomingTransactionHelper.mjs
+++ b/dist/helpers/IncomingTransactionHelper.mjs
@@ -14,7 +14,7 @@ var _IncomingTransactionHelper_instances, _IncomingTransactionHelper_getCache, _
 // eslint-disable-next-line import-x/no-nodejs-modules
 import EventEmitter from "events";
 import { incomingTransactionsLogger as log } from "../logger.mjs";
-const INTERVAL = 1000 * 30; // 30 Seconds
+const INTERVAL = 1000 * 60 * 4; // 4 Minutes
 export class IncomingTransactionHelper {
     constructor({ getCache, getCurrentAccount, getLocalTransactions, includeTokenTransfers, isEnabled, queryEntireHistory, remoteTransactionSource, trimTransactions, updateCache, updateTransactions, }) {
         _IncomingTransactionHelper_instances.add(this);
