diff --git a/dist/snaps/SnapController.cjs b/dist/snaps/SnapController.cjs
index 1699ba7939a87cfd2776490be8e8a26b1413279d..d5ed95d9d786b3807575a7c75c308417cfe4c52a 100644
--- a/dist/snaps/SnapController.cjs
+++ b/dist/snaps/SnapController.cjs
@@ -523,6 +523,7 @@ class SnapController extends base_controller_1.BaseController {
         this.update((state) => {
             state.snaps = {};
             state.snapStates = {};
+            state.unencryptedSnapStates = {};
         });
         __classPrivateFieldGet(this, _SnapController_snapsRuntimeData, "f").clear();
         // We want to remove all snaps & permissions, except for preinstalled snaps
diff --git a/dist/snaps/SnapController.mjs b/dist/snaps/SnapController.mjs
index b52ee05c9c4d83bc7b27e76e6229efc032b0146c..d1dfc88a95ebce026d292223bae866b361b5630c 100644
--- a/dist/snaps/SnapController.mjs
+++ b/dist/snaps/SnapController.mjs
@@ -520,6 +520,7 @@ export class SnapController extends BaseController {
         this.update((state) => {
             state.snaps = {};
             state.snapStates = {};
+            state.unencryptedSnapStates = {};
         });
         __classPrivateFieldGet(this, _SnapController_snapsRuntimeData, "f").clear();
         // We want to remove all snaps & permissions, except for preinstalled snaps
