diff --git a/dist/SmartTransactionsController.js b/dist/SmartTransactionsController.js
index cd9bd16a3b0ddbb49de6bc7bd8fceefb8b56ad5e..066e2e132e55f2dcc9a8add4197b8697c2ff4d90 100644
--- a/dist/SmartTransactionsController.js
+++ b/dist/SmartTransactionsController.js
@@ -77,8 +77,8 @@ class SmartTransactionsController extends polling_controller_1.StaticIntervalPol
         this.getMetaMetricsProps = getMetaMetricsProps;
         this.initializeSmartTransactionsForChainId();
         __classPrivateFieldGet(this, _SmartTransactionsController_instances, "m", _SmartTransactionsController_ensureUniqueSmartTransactions).call(this);
-        onNetworkStateChange(({ providerConfig: newProvider }) => {
-            const { chainId } = newProvider;
+        onNetworkStateChange((state) => {
+            const { chainId } = this.getNetworkClientById(state.selectedNetworkClientId);
             const isNewChainId = chainId !== this.config.chainId;
             this.configure({ chainId });
             this.initializeSmartTransactionsForChainId();
