diff --git a/dist/assetsUtil.cjs b/dist/assetsUtil.cjs
index 2c759f2ecc76b0a6f467166a2e4d077dc951445e..520b22b5ff48a955d5e29dc4259f3e9eb1f086ff 100644
--- a/dist/assetsUtil.cjs
+++ b/dist/assetsUtil.cjs
@@ -237,7 +237,8 @@ async function getIpfsCIDv1AndPath(ipfsUrl) {
     const index = url.indexOf('/');
     const cid = index !== -1 ? url.substring(0, index) : url;
     const path = index !== -1 ? url.substring(index) : undefined;
-    const { CID } = await import("multiformats");
+    // Use CJS require to allow Jest to parse without ESM transform
+    const CID = require("multiformats").CID;
     // We want to ensure that the CID is v1 (https://docs.ipfs.io/concepts/content-addressing/#identifier-formats)
     // because most cid v0s appear to be incompatible with IPFS subdomains
     return {
