diff --git a/dist/KeyringController.js b/dist/KeyringController.js
index da3fb9c0bf437e07f1927bca0bbfc94ed8454c2d..a3211dc83edb5d18daf328adc5a63fee920dd321 100644
--- a/dist/KeyringController.js
+++ b/dist/KeyringController.js
@@ -713,7 +713,6 @@ class KeyringController extends base_controller_1.BaseController {
                     throw new Error('Seed phrase imported different accounts.');
                 }
             });
-            return seedWords;
         });
     }
     // QR Hardware related methods
