diff --git a/dist/TransactionController.cjs b/dist/TransactionController.cjs
index c96d2962d81c4c63c7b626bc1ca18b196b20a154..236b451e5f94584d4ce42b6bfaf6b7225b8ae7f9 100644
--- a/dist/TransactionController.cjs
+++ b/dist/TransactionController.cjs
@@ -1152,7 +1152,11 @@ class TransactionController extends base_controller_1.BaseController {
                 if (!isTxCompleted) {
                     if (error?.code === rpc_errors_1.errorCodes.provider.userRejectedRequest) {
                         this.cancelTransaction(transactionId, actionId);
-                        throw rpc_errors_1.providerErrors.userRejectedRequest('MetaMask Tx Signature: User denied transaction signature.');
+                        throw rpc_errors_1.providerErrors.userRejectedRequest({
+                            message: 
+                                'MetaMask Tx Signature: User denied transaction signature.',
+                            data: error?.data,
+                        });
                     }
                     else {
                         this.failTransaction(meta, error, actionId);
