diff --git a/dist/TokenBalancesController.cjs b/dist/TokenBalancesController.cjs
index 4918812dde60b8d0e24a7bded27d88f233968858..4e8018bce92b9e5d47fc40784409e16db22be615 100644
--- a/dist/TokenBalancesController.cjs
+++ b/dist/TokenBalancesController.cjs
@@ -535,14 +535,16 @@ class TokenBalancesController extends (0, polling_controller_1.StaticIntervalPol
             }
             // Update with actual fetched balances only if the value has changed
             aggregated.forEach(({ success, value, account, token, chainId }) => {
-                var _a, _b, _c;
+                var _a, _b;
                 if (success && value !== undefined) {
+                    // Ensure all accounts we add/update are in lower-case
+                    const lowerCaseAccount = account.toLowerCase();
                     const newBalance = (0, controller_utils_1.toHex)(value);
                     const tokenAddress = checksum(token);
-                    const currentBalance = d.tokenBalances[account]?.[chainId]?.[tokenAddress];
+                    const currentBalance = d.tokenBalances[lowerCaseAccount]?.[chainId]?.[tokenAddress];
                     // Only update if the balance has actually changed
                     if (currentBalance !== newBalance) {
-                        ((_c = ((_a = d.tokenBalances)[_b = account] ?? (_a[_b] = {})))[chainId] ?? (_c[chainId] = {}))[tokenAddress] = newBalance;
+                        ((_b = ((_a = d.tokenBalances)[lowerCaseAccount] ?? (_a[lowerCaseAccount] = {})))[chainId] ?? (_b[chainId] = {}))[tokenAddress] = newBalance;
                     }
                 }
             });
diff --git a/dist/TokenBalancesController.mjs b/dist/TokenBalancesController.mjs
index f64d13f8de56631345a44e6ebb025e62e03f51bc..99aa7f27c574c94b26daa56091ac50d15281dd30 100644
--- a/dist/TokenBalancesController.mjs
+++ b/dist/TokenBalancesController.mjs
@@ -531,14 +531,16 @@ export class TokenBalancesController extends StaticIntervalPollingController() {
             }
             // Update with actual fetched balances only if the value has changed
             aggregated.forEach(({ success, value, account, token, chainId }) => {
-                var _a, _b, _c;
+                var _a, _b;
                 if (success && value !== undefined) {
+                    // Ensure all accounts we add/update are in lower-case
+                    const lowerCaseAccount = account.toLowerCase();
                     const newBalance = toHex(value);
                     const tokenAddress = checksum(token);
-                    const currentBalance = d.tokenBalances[account]?.[chainId]?.[tokenAddress];
+                    const currentBalance = d.tokenBalances[lowerCaseAccount]?.[chainId]?.[tokenAddress];
                     // Only update if the balance has actually changed
                     if (currentBalance !== newBalance) {
-                        ((_c = ((_a = d.tokenBalances)[_b = account] ?? (_a[_b] = {})))[chainId] ?? (_c[chainId] = {}))[tokenAddress] = newBalance;
+                        ((_b = ((_a = d.tokenBalances)[lowerCaseAccount] ?? (_a[lowerCaseAccount] = {})))[chainId] ?? (_b[chainId] = {}))[tokenAddress] = newBalance;
                     }
                 }
             });
diff --git a/dist/token-prices-service/codefi-v2.cjs b/dist/token-prices-service/codefi-v2.cjs
index ba0f0c1bcbf0f231549b1ca9d3be2d1137a0d732..fd4851471fa0c2f07efbb527a3eea55cbfbc4743 100644
--- a/dist/token-prices-service/codefi-v2.cjs
+++ b/dist/token-prices-service/codefi-v2.cjs
@@ -100,6 +100,8 @@ exports.SUPPORTED_CURRENCIES = [
     'mxn',
     // Malaysian Ringgit
     'myr',
+    // Monad
+    'mon',
     // Nigerian Naira
     'ngn',
     // Norwegian Krone
@@ -220,43 +222,43 @@ exports.getNativeTokenAddress = getNativeTokenAddress;
 // Source: https://github.com/consensys-vertical-apps/va-mmcx-price-api/blob/main/src/constants/slip44.ts
 // We can only support PricesAPI V3 for EVM chains that have a CAIP-19 native asset mapping.
 exports.SPOT_PRICES_SUPPORT_INFO = {
-    '0x1': 'eip155:1/slip44:60', // Ethereum Mainnet - Native symbol: ETH
-    '0xa': 'eip155:10/slip44:60', // OP Mainnet - Native symbol: ETH
-    '0x19': 'eip155:25/slip44:394', // Cronos Mainnet - Native symbol: CRO
-    '0x38': 'eip155:56/slip44:714', // BNB Smart Chain Mainnet - Native symbol: BNB
-    '0x39': 'eip155:57/erc20:0x0000000000000000000000000000000000000000', // 'eip155:57/slip44:57', // Syscoin Mainnet - Native symbol: SYS
+    '0x1': null, //'eip155:1/slip44:60', // Ethereum Mainnet - Native symbol: ETH
+    '0xa': null, //'eip155:10/slip44:60', // OP Mainnet - Native symbol: ETH
+    '0x19': null, //'eip155:25/slip44:394', // Cronos Mainnet - Native symbol: CRO
+    '0x38': null, //'eip155:56/slip44:714', // BNB Smart Chain Mainnet - Native symbol: BNB
+    '0x39': null, //'eip155:57/erc20:0x0000000000000000000000000000000000000000', // 'eip155:57/slip44:57', // Syscoin Mainnet - Native symbol: SYS
     '0x52': null, // 'eip155:82/slip44:18000', // Meter Mainnet - Native symbol: MTR
-    '0x58': 'eip155:88/erc20:0x0000000000000000000000000000000000000000', // 'eip155:88/slip44:889', // TomoChain - Native symbol: TOMO
-    '0x64': 'eip155:100/slip44:700', // Gnosis (formerly xDAI Chain) - Native symbol: xDAI
-    '0x6a': 'eip155:106/erc20:0x0000000000000000000000000000000000000000', // 'eip155:106/slip44:5655640', // Velas EVM Mainnet - Native symbol: VLX
-    '0x80': 'eip155:128/erc20:0x0000000000000000000000000000000000000000', // 'eip155:128/slip44:1010', // Huobi ECO Chain Mainnet - Native symbol: HT
-    '0x89': 'eip155:137/slip44:966', // Polygon Mainnet - Native symbol: POL
+    '0x58': null, //'eip155:88/erc20:0x0000000000000000000000000000000000000000', // 'eip155:88/slip44:889', // TomoChain - Native symbol: TOMO
+    '0x64': null, //'eip155:100/slip44:700', // Gnosis (formerly xDAI Chain) - Native symbol: xDAI
+    '0x6a': null, //'eip155:106/erc20:0x0000000000000000000000000000000000000000', // 'eip155:106/slip44:5655640', // Velas EVM Mainnet - Native symbol: VLX
+    '0x80': null, //'eip155:128/erc20:0x0000000000000000000000000000000000000000', // 'eip155:128/slip44:1010', // Huobi ECO Chain Mainnet - Native symbol: HT
+    '0x89': null, //'eip155:137/slip44:966', // Polygon Mainnet - Native symbol: POL
     '0x8f': null, // 'eip155:143/slip44:268435779', // Monad Mainnet - Native symbol: MON
-    '0x92': 'eip155:146/slip44:10007', // Sonic Mainnet - Native symbol: S
-    '0xfa': 'eip155:250/slip44:1007', // Fantom Opera - Native symbol: FTM
-    '0x141': 'eip155:321/erc20:0x0000000000000000000000000000000000000000', // 'eip155:321/slip44:641', // KCC Mainnet - Native symbol: KCS
-    '0x144': 'eip155:324/slip44:60', // zkSync Era Mainnet (Ethereum L2) - Native symbol: ETH
-    '0x169': 'eip155:361/erc20:0x0000000000000000000000000000000000000000', // 'eip155:361/slip44:589', // Theta Mainnet - Native symbol: TFUEL
-    '0x3e7': 'eip155:999/slip44:2457', // HyperEVM - Native symbol: ETH
-    '0x440': 'eip155:1088/erc20:0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000', // 'eip155:1088/slip44:XXX', // Metis Andromeda Mainnet (Ethereum L2) - Native symbol: METIS
-    '0x44d': 'eip155:1101/slip44:60', // Polygon zkEVM mainnet - Native symbol: ETH
-    '0x504': 'eip155:1284/slip44:1284', // Moonbeam - Native symbol: GLMR
-    '0x505': 'eip155:1285/slip44:1285', // Moonriver - Native symbol: MOVR
-    '0x531': 'eip155:1329/slip44:19000118', // Sei Mainnet - Native symbol: SEI
-    '0x1388': 'eip155:5000/erc20:0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000', // 'eip155:5000/slip44:XXX', // Mantle - Native symbol: MNT
-    '0x2105': 'eip155:8453/slip44:60', // Base - Native symbol: ETH
-    '0x2710': 'eip155:10000/erc20:0x0000000000000000000000000000000000000000', // 'eip155:10000/slip44:145', // Smart Bitcoin Cash - Native symbol: BCH
-    '0xa4b1': 'eip155:42161/slip44:60', // Arbitrum One - Native symbol: ETH
-    '0xa4ec': 'eip155:42220/slip44:52752', // Celo Mainnet - Native symbol: CELO
-    '0xa516': 'eip155:42262/erc20:0x0000000000000000000000000000000000000000', // 'eip155:42262/slip44:474', // Oasis Emerald - Native symbol: ROSE
-    '0xa86a': 'eip155:43114/slip44:9005', // Avalanche C-Chain - Native symbol: AVAX
-    '0xe708': 'eip155:59144/slip44:60', // Linea Mainnet - Native symbol: ETH
-    '0x13c31': 'eip155:81457/erc20:0x0000000000000000000000000000000000000000', // 'eip155:81457/slip44:60', // Blast Mainnet - Native symbol: ETH
-    '0x17dcd': 'eip155:97741/erc20:0x0000000000000000000000000000000000000000', // 'eip155:97741/slip44:XXX', // Pepe Unchained Mainnet - Native symbol: PEPU
+    '0x92': null, //'eip155:146/slip44:10007', // Sonic Mainnet - Native symbol: S
+    '0xfa': null, //'eip155:250/slip44:1007', // Fantom Opera - Native symbol: FTM
+    '0x141': null, //'eip155:321/erc20:0x0000000000000000000000000000000000000000', // 'eip155:321/slip44:641', // KCC Mainnet - Native symbol: KCS
+    '0x144': null, //'eip155:324/slip44:60', // zkSync Era Mainnet (Ethereum L2) - Native symbol: ETH
+    '0x169': null, //'eip155:361/erc20:0x0000000000000000000000000000000000000000', // 'eip155:361/slip44:589', // Theta Mainnet - Native symbol: TFUEL
+    '0x3e7': null, //'eip155:999/slip44:2457', // HyperEVM - Native symbol: ETH
+    '0x440': null, //'eip155:1088/erc20:0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000', // 'eip155:1088/slip44:XXX', // Metis Andromeda Mainnet (Ethereum L2) - Native symbol: METIS
+    '0x44d': null, //'eip155:1101/slip44:60', // Polygon zkEVM mainnet - Native symbol: ETH
+    '0x504': null, //'eip155:1284/slip44:1284', // Moonbeam - Native symbol: GLMR
+    '0x505': null, //'eip155:1285/slip44:1285', // Moonriver - Native symbol: MOVR
+    '0x531': null, //'eip155:1329/slip44:19000118', // Sei Mainnet - Native symbol: SEI
+    '0x1388': null, //'eip155:5000/erc20:0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000', // 'eip155:5000/slip44:XXX', // Mantle - Native symbol: MNT
+    '0x2105': null, //'eip155:8453/slip44:60', // Base - Native symbol: ETH
+    '0x2710': null, //'eip155:10000/erc20:0x0000000000000000000000000000000000000000', // 'eip155:10000/slip44:145', // Smart Bitcoin Cash - Native symbol: BCH
+    '0xa4b1': null, //'eip155:42161/slip44:60', // Arbitrum One - Native symbol: ETH
+    '0xa4ec': null, //'eip155:42220/slip44:52752', // Celo Mainnet - Native symbol: CELO
+    '0xa516': null, //'eip155:42262/erc20:0x0000000000000000000000000000000000000000', // 'eip155:42262/slip44:474', // Oasis Emerald - Native symbol: ROSE
+    '0xa86a': null, //'eip155:43114/slip44:9005', // Avalanche C-Chain - Native symbol: AVAX
+    '0xe708': null, //'eip155:59144/slip44:60', // Linea Mainnet - Native symbol: ETH
+    '0x13c31': null, //'eip155:81457/erc20:0x0000000000000000000000000000000000000000', // 'eip155:81457/slip44:60', // Blast Mainnet - Native symbol: ETH
+    '0x17dcd': null, //'eip155:97741/erc20:0x0000000000000000000000000000000000000000', // 'eip155:97741/slip44:XXX', // Pepe Unchained Mainnet - Native symbol: PEPU
     '0x518af': null, // 'eip155:333999/slip44:1997', // Polis Mainnet - Native symbol: POLIS
-    '0x82750': 'eip155:534352/slip44:60', // Scroll Mainnet - Native symbol: ETH
-    '0x4e454152': 'eip155:60/slip44:60', // Aurora Mainnet (Ethereum L2 on NEAR) - Native symbol: ETH
-    '0x63564c40': 'eip155:1666600000/slip44:1023', // Harmony Mainnet Shard 0 - Native symbol: ONE
+    '0x82750': null, //'eip155:534352/slip44:60', // Scroll Mainnet - Native symbol: ETH
+    '0x4e454152': null, //'eip155:60/slip44:60', // Aurora Mainnet (Ethereum L2 on NEAR) - Native symbol: ETH
+    '0x63564c40': null, //'eip155:1666600000/slip44:1023', // Harmony Mainnet Shard 0 - Native symbol: ONE
 };
 /**
  * The list of chain IDs that can be supplied in the URL for the `/spot-prices`
