diff --git a/dist/bridge-status-controller.cjs b/dist/bridge-status-controller.cjs
index 84e7c4816afba3765e9e30e9fd403b20e6883723..1c1eaef962391a8add1db122ad418ed9e12e4503 100644
--- a/dist/bridge-status-controller.cjs
+++ b/dist/bridge-status-controller.cjs
@@ -295,11 +295,7 @@ class BridgeStatusController extends (0, polling_controller_1.StaticIntervalPoll
             if (!selectedAccount?.metadata?.snap?.id) {
                 throw new Error('Failed to submit cross-chain swap transaction: undefined snap id');
             }
-            const bridgeFeatureFlags = (0, bridge_controller_1.getBridgeFeatureFlags)(this.messagingSystem);
-            const request = bridgeFeatureFlags?.chains?.[keyring_api_1.SolScope.Mainnet]
-                ?.isSnapConfirmationEnabled
-                ? (0, transaction_1.getKeyringRequest)(quoteResponse, selectedAccount)
-                : (0, transaction_1.getClientRequest)(quoteResponse, selectedAccount);
+            const request = (0, transaction_1.getClientRequest)(quoteResponse, selectedAccount);
             const requestResponse = (await this.messagingSystem.call('SnapController:handleRequest', request));
             // The extension client actually redirects before it can do anytyhing with this meta
             const txMeta = (0, transaction_1.handleSolanaTxResponse)(requestResponse, quoteResponse, selectedAccount);
diff --git a/dist/bridge-status-controller.mjs b/dist/bridge-status-controller.mjs
index d1c76aaf9bffca698520f5c8903f1c8151064f3b..aee7d633a2a4eadab73c2812fd7e30f8fe57a4e7 100644
--- a/dist/bridge-status-controller.mjs
+++ b/dist/bridge-status-controller.mjs
@@ -292,11 +292,7 @@ export class BridgeStatusController extends StaticIntervalPollingController() {
             if (!selectedAccount?.metadata?.snap?.id) {
                 throw new Error('Failed to submit cross-chain swap transaction: undefined snap id');
             }
-            const bridgeFeatureFlags = getBridgeFeatureFlags(this.messagingSystem);
-            const request = bridgeFeatureFlags?.chains?.[SolScope.Mainnet]
-                ?.isSnapConfirmationEnabled
-                ? getKeyringRequest(quoteResponse, selectedAccount)
-                : getClientRequest(quoteResponse, selectedAccount);
+            const request = getClientRequest(quoteResponse, selectedAccount);
             const requestResponse = (await this.messagingSystem.call('SnapController:handleRequest', request));
             // The extension client actually redirects before it can do anytyhing with this meta
             const txMeta = handleSolanaTxResponse(requestResponse, quoteResponse, selectedAccount);
