diff --git a/dist/utils/transaction.cjs b/dist/utils/transaction.cjs
index 0db5e5bf4815125dc6839e1ee6786fe041c85ab1..df594ba084bb31c6332ca6d1d4d05e5c8b87d657 100644
--- a/dist/utils/transaction.cjs
+++ b/dist/utils/transaction.cjs
@@ -80,10 +80,7 @@ function subscribeAssetChanges(messenger, getControllerState, updateTransactionD
             onTransactionChange(transaction, messenger, updateTransactionData);
         }
     };
-    if ((0, feature_flags_1.getAssetsUnifyStateFeature)(messenger)) {
-        messenger.subscribe('AssetsController:stateChange', buildHandler('AssetsController'));
-        return;
-    }
+    messenger.subscribe('AssetsController:stateChange', buildHandler('AssetsController'));
     messenger.subscribe('TokensController:stateChange', buildHandler('TokensController'));
     messenger.subscribe('TokenRatesController:stateChange', buildHandler('TokenRatesController'));
     messenger.subscribe('CurrencyRateController:stateChange', buildHandler('CurrencyRateController'));
diff --git a/dist/utils/transaction.mjs b/dist/utils/transaction.mjs
index b778c91a239d76b990f814735d2c902ee478af96..13907c6a9f169f66587b02cab9f40ecd33e33256 100644
--- a/dist/utils/transaction.mjs
+++ b/dist/utils/transaction.mjs
@@ -76,10 +76,7 @@ export function subscribeAssetChanges(messenger, getControllerState, updateTrans
             onTransactionChange(transaction, messenger, updateTransactionData);
         }
     };
-    if (getAssetsUnifyStateFeature(messenger)) {
-        messenger.subscribe('AssetsController:stateChange', buildHandler('AssetsController'));
-        return;
-    }
+    messenger.subscribe('AssetsController:stateChange', buildHandler('AssetsController'));
     messenger.subscribe('TokensController:stateChange', buildHandler('TokensController'));
     messenger.subscribe('TokenRatesController:stateChange', buildHandler('TokenRatesController'));
     messenger.subscribe('CurrencyRateController:stateChange', buildHandler('CurrencyRateController'));
