diff --git a/dist/NetworkController.cjs b/dist/NetworkController.cjs
index b85de6f29620feecf6f1e395f3d25c96fb10f204..112d61761d000906d4fa7b4ace687ac67f501b9e 100644
--- a/dist/NetworkController.cjs
+++ b/dist/NetworkController.cjs
@@ -408,7 +408,7 @@ class NetworkController extends base_controller_1.BaseController {
      */
     async initializeProvider() {
         __classPrivateFieldGet(this, _NetworkController_instances, "m", _NetworkController_applyNetworkSelection).call(this, this.state.selectedNetworkClientId);
-        await this.lookupNetwork();
+        // await this.lookupNetwork();
     }
     /**
      * Refreshes the network meta with EIP-1559 support and the network status
diff --git a/dist/NetworkController.mjs b/dist/NetworkController.mjs
index 966da694324513857dc6766590a4c965dedc2e3d..e268f1fa8f5461fc4d00441823eef3dfd3eff7d7 100644
--- a/dist/NetworkController.mjs
+++ b/dist/NetworkController.mjs
@@ -383,7 +383,7 @@ export class NetworkController extends BaseController {
      */
     async initializeProvider() {
         __classPrivateFieldGet(this, _NetworkController_instances, "m", _NetworkController_applyNetworkSelection).call(this, this.state.selectedNetworkClientId);
-        await this.lookupNetwork();
+        // await this.lookupNetwork();
     }
     /**
      * Refreshes the network meta with EIP-1559 support and the network status
