diff --git a/dist/cjs/restricted/dialog.js b/dist/cjs/restricted/dialog.js
index 037a374ee23d24143975708669253cc99da0b3f7..22aa902c3947ae1379df51734f6cecdd958d3acb 100644
--- a/dist/cjs/restricted/dialog.js
+++ b/dist/cjs/restricted/dialog.js
@@ -104,8 +104,6 @@ function getDialogImplementation({ showDialog, isOnPhishingList, maybeUpdatePhis
         const validatedType = getValidatedType(params);
         const validatedParams = getValidatedParams(params, structs[validatedType]);
         const { content } = validatedParams;
-        await maybeUpdatePhishingList();
-        (0, _snapsui.assertUILinksAreSafe)(content, isOnPhishingList);
         const placeholder = validatedParams.type === DialogType.Prompt ? validatedParams.placeholder : undefined;
         return showDialog(origin, validatedType, content, placeholder);
     };
diff --git a/dist/cjs/restricted/notify.js b/dist/cjs/restricted/notify.js
index c9d9b7e3a250919ffcf4cc8f97e8c11fe9d272b6..c5d9fb0d833777eb3aa7d57071dade116373fd67 100644
--- a/dist/cjs/restricted/notify.js
+++ b/dist/cjs/restricted/notify.js
@@ -61,8 +61,6 @@ function getImplementation({ showNativeNotification, showInAppNotification, isOn
     return async function implementation(args) {
         const { params, context: { origin } } = args;
         const validatedParams = getValidatedParams(params);
-        await maybeUpdatePhishingList();
-        (0, _snapsui.assertLinksAreSafe)(validatedParams.message, isOnPhishingList);
         switch(validatedParams.type){
             case NotificationType.Native:
                 return await showNativeNotification(origin, validatedParams);
diff --git a/dist/esm/restricted/dialog.js b/dist/esm/restricted/dialog.js
index 018801436c3675c5a1f6fa4647eaaaeb6847c90f..21c49b73584bd4cccd892cd4cdf2459695e4f064 100644
--- a/dist/esm/restricted/dialog.js
+++ b/dist/esm/restricted/dialog.js
@@ -94,8 +94,6 @@ const structs = {
         const validatedType = getValidatedType(params);
         const validatedParams = getValidatedParams(params, structs[validatedType]);
         const { content } = validatedParams;
-        await maybeUpdatePhishingList();
-        assertUILinksAreSafe(content, isOnPhishingList);
         const placeholder = validatedParams.type === DialogType.Prompt ? validatedParams.placeholder : undefined;
         return showDialog(origin, validatedType, content, placeholder);
     };
diff --git a/dist/esm/restricted/notify.js b/dist/esm/restricted/notify.js
index b33d163d6fb8741c05d5a6acc0ef3735c239e11b..51673869dc50e764f2ce7ee77838adba8dd8e5f6 100644
--- a/dist/esm/restricted/notify.js
+++ b/dist/esm/restricted/notify.js
@@ -52,8 +52,6 @@ export const notifyBuilder = Object.freeze({
     return async function implementation(args) {
         const { params, context: { origin } } = args;
         const validatedParams = getValidatedParams(params);
-        await maybeUpdatePhishingList();
-        assertLinksAreSafe(validatedParams.message, isOnPhishingList);
         switch(validatedParams.type){
             case NotificationType.Native:
                 return await showNativeNotification(origin, validatedParams);
