diff --git a/dist/utils.js b/dist/utils.js
index 810f229841ffff83f7a28191bc558862b1809e01..31047fd3387693f30a76807d596a8de86bb505e1 100644
--- a/dist/utils.js
+++ b/dist/utils.js
@@ -11,6 +11,12 @@ const uuid_1 = require("uuid");
  * @returns The name of the keyring type.
  */
 function keyringTypeToName(keyringType) {
+
+    // Custody Keyrings follow this pattern
+    if (/^Custody.*/.test(keyringType)) {
+        return "Custody";
+    }
+
     switch (keyringType) {
         case keyring_controller_1.KeyringTypes.simple: {
             return 'Account';
