diff --git a/dist/chunk-HBNET6MY.mjs b/dist/chunk-HBNET6MY.mjs
index 5843d635c91e9125d130a6ef5688550e5d514eaa..6ef8d23c52c7ddb28a1912967e4b658fe6f8ef1c 100644
--- a/dist/chunk-HBNET6MY.mjs
+++ b/dist/chunk-HBNET6MY.mjs
@@ -1749,7 +1749,7 @@ decryptSnapState_fn = async function(snapId, state) {
       snapId,
       salt,
       useCache,
-      keyMetadata
+      keyMetadata: keyMetadata ?? { algorithm: 'PBKDF2', params: { iterations: 10_000 } }
     });
     const decryptedState = await __privateGet(this, _encryptor).decryptWithKey(key, parsed);
     assert(isValidJson(decryptedState));
diff --git a/dist/chunk-JMP6XYRL.js b/dist/chunk-JMP6XYRL.js
index 20c3b12643e4f87d70aa0df96a36c4dda8b1ed47..c442feb729f41d9d73d1008a3215d73d7b288090 100644
--- a/dist/chunk-JMP6XYRL.js
+++ b/dist/chunk-JMP6XYRL.js
@@ -1749,7 +1749,7 @@ decryptSnapState_fn = async function(snapId, state) {
       snapId,
       salt,
       useCache,
-      keyMetadata
+      keyMetadata: keyMetadata ?? { algorithm: 'PBKDF2', params: { iterations: 10_000 } }
     });
     const decryptedState = await _chunkEXN2TFDJjs.__privateGet.call(void 0, this, _encryptor).decryptWithKey(key, parsed);
     _utils.assert.call(void 0, _utils.isValidJson.call(void 0, decryptedState));
