diff --git a/dist/selectors.cjs b/dist/selectors.cjs
index 35c1970c214bcc26944cdfb903919815ee7d3382..2c03ef1314faf8a9451808ce92721d130a8d5e50 100644
--- a/dist/selectors.cjs
+++ b/dist/selectors.cjs
@@ -224,6 +224,9 @@ const selectSortedBridgeQuotes = createBridgeSelector([
         case types_js_1.SortOrder.ETA_ASC:
             return (0, lodash_1.orderBy)(quotesWithMetadata, (quote) => quote.estimatedProcessingTimeInSeconds, 'asc');
         default:
+            if (quotesWithMetadata.every((quote) => quote.cost?.valueInCurrency)) {
+                return (0, lodash_1.orderBy)(quotesWithMetadata, ({ cost }) => Number(cost?.valueInCurrency), 'asc');
+            }
             if (quotesWithMetadata.every((quote) => quote.quote.priceData?.priceImpact?.amount)) {
                 return (0, lodash_1.orderBy)(quotesWithMetadata, ({ quote: { priceData } }) => Number(priceData?.priceImpact?.amount), 'asc');
             }
diff --git a/dist/selectors.mjs b/dist/selectors.mjs
index a9aa4edde11816067f9547d1fd5ee2379c759c09..ade7ec49a41a613acf2f4966bd9652d1780df887 100644
--- a/dist/selectors.mjs
+++ b/dist/selectors.mjs
@@ -220,6 +220,9 @@ const selectSortedBridgeQuotes = createBridgeSelector([
         case SortOrder.ETA_ASC:
             return orderBy(quotesWithMetadata, (quote) => quote.estimatedProcessingTimeInSeconds, 'asc');
         default:
+            if (quotesWithMetadata.every((quote) => quote.cost?.valueInCurrency)) {
+                return orderBy(quotesWithMetadata, ({ cost }) => Number(cost?.valueInCurrency), 'asc');
+            }
             if (quotesWithMetadata.every((quote) => quote.quote.priceData?.priceImpact?.amount)) {
                 return orderBy(quotesWithMetadata, ({ quote: { priceData } }) => Number(priceData?.priceImpact?.amount), 'asc');
             }
