diff --git a/dist/chunk-6W5EQ3JQ.js b/dist/chunk-6W5EQ3JQ.js
index 0dc70ec5f77484f8d4021bdfff7bca90dd87bd51..81c2d0d1e6a0ad6db8b361423774b9a843fb46f6 100644
--- a/dist/chunk-6W5EQ3JQ.js
+++ b/dist/chunk-6W5EQ3JQ.js
@@ -380,7 +380,7 @@ fetchAndMapExchangeRatesForSupportedNativeCurrency_fn = async function({
     (obj, [tokenAddress, token]) => {
       obj = {
         ...obj,
-        [tokenAddress.toLowerCase()]: { ...token }
+        [tokenAddress]: { ...token }
       };
       return obj;
     },
