diff --git a/dist/NetworkController.js b/dist/NetworkController.js
index 4bce7f59a13e31d775582c4aca5016500f40667b..b8b4e5eeb6da9012e7d929850e4c90b22b91b451 100644
--- a/dist/NetworkController.js
+++ b/dist/NetworkController.js
@@ -298,11 +298,14 @@ class NetworkController extends base_controller_1.BaseControllerV2 {
      * Populates the network clients and establishes the initial network based on
      * the provider configuration in state.
      */
-    initializeProvider() {
+    initializeProvider(lookup) {
         return __awaiter(this, void 0, void 0, function* () {
+            lookup = lookup !== undefined ? lookup : true;
             __classPrivateFieldGet(this, _NetworkController_instances, "m", _NetworkController_ensureAutoManagedNetworkClientRegistryPopulated).call(this);
             __classPrivateFieldGet(this, _NetworkController_instances, "m", _NetworkController_applyNetworkSelection).call(this);
-            yield this.lookupNetwork();
+            if (lookup) {
+                yield this.lookupNetwork();
+            }
         });
     }
     /**
