diff --git a/dist/NotificationServicesController/NotificationServicesController.cjs b/dist/NotificationServicesController/NotificationServicesController.cjs
index c5283c8fff2b58787d654c2a4a700ff212732d27..2c0e69468c3bab825ca6d9e6d78d9fb3b4372081 100644
--- a/dist/NotificationServicesController/NotificationServicesController.cjs
+++ b/dist/NotificationServicesController/NotificationServicesController.cjs
@@ -237,12 +237,14 @@ class NotificationServicesController extends base_controller_1.BaseController {
             // Flag to ensure we only setup once
             isNotificationAccountsSetup: false,
             getNotificationAccounts: async () => {
-                const mainHDWalletAccounts = (await this.messagingSystem.call('KeyringController:withKeyring', {
+                const mainHDWalletAccounts = (await this.messagingSystem
+                    .call('KeyringController:withKeyring', {
                     type: keyring_controller_1.KeyringTypes.hd,
                     index: 0,
                 }, async ({ keyring }) => {
                     return await keyring.getAccounts();
-                }));
+                })
+                    .catch(() => null));
                 return mainHDWalletAccounts;
             },
             /**
@@ -253,6 +255,13 @@ class NotificationServicesController extends base_controller_1.BaseController {
             listAccounts: async () => {
                 // Get previous and current account sets
                 const nonChecksumAccounts = await __classPrivateFieldGet(this, _NotificationServicesController_accounts, "f").getNotificationAccounts();
+                if (!nonChecksumAccounts) {
+                    return {
+                        accountsAdded: [],
+                        accountsRemoved: [],
+                        accounts: [],
+                    };
+                }
                 const accounts = nonChecksumAccounts
                     .map((a) => (0, controller_utils_1.toChecksumHexAddress)(a))
                     .filter((a) => (0, controller_utils_1.isValidHexAddress)(a));
diff --git a/dist/NotificationServicesController/NotificationServicesController.mjs b/dist/NotificationServicesController/NotificationServicesController.mjs
index 6c7ac433051bf0f38da6c367c25a3fc6bf8f948e..1eeede947b3a8875589a81c140f001f358d8bcc1 100644
--- a/dist/NotificationServicesController/NotificationServicesController.mjs
+++ b/dist/NotificationServicesController/NotificationServicesController.mjs
@@ -215,12 +215,14 @@ class NotificationServicesController extends BaseController {
             // Flag to ensure we only setup once
             isNotificationAccountsSetup: false,
             getNotificationAccounts: async () => {
-                const mainHDWalletAccounts = (await this.messagingSystem.call('KeyringController:withKeyring', {
+                const mainHDWalletAccounts = (await this.messagingSystem
+                    .call('KeyringController:withKeyring', {
                     type: KeyringTypes.hd,
                     index: 0,
                 }, async ({ keyring }) => {
                     return await keyring.getAccounts();
-                }));
+                })
+                    .catch(() => null));
                 return mainHDWalletAccounts;
             },
             /**
@@ -231,6 +233,13 @@ class NotificationServicesController extends BaseController {
             listAccounts: async () => {
                 // Get previous and current account sets
                 const nonChecksumAccounts = await __classPrivateFieldGet(this, _NotificationServicesController_accounts, "f").getNotificationAccounts();
+                if (!nonChecksumAccounts) {
+                    return {
+                        accountsAdded: [],
+                        accountsRemoved: [],
+                        accounts: [],
+                    };
+                }
                 const accounts = nonChecksumAccounts
                     .map((a) => toChecksumHexAddress(a))
                     .filter((a) => isValidHexAddress(a));
