diff --git a/dist/TokenBalancesController.cjs b/dist/TokenBalancesController.cjs
index 4918812dde60b8d0e24a7bded27d88f233968858..4e8018bce92b9e5d47fc40784409e16db22be615 100644
--- a/dist/TokenBalancesController.cjs
+++ b/dist/TokenBalancesController.cjs
@@ -535,14 +535,16 @@ class TokenBalancesController extends (0, polling_controller_1.StaticIntervalPol
             }
             // Update with actual fetched balances only if the value has changed
             aggregated.forEach(({ success, value, account, token, chainId }) => {
-                var _a, _b, _c;
+                var _a, _b;
                 if (success && value !== undefined) {
+                    // Ensure all accounts we add/update are in lower-case
+                    const lowerCaseAccount = account.toLowerCase();
                     const newBalance = (0, controller_utils_1.toHex)(value);
                     const tokenAddress = checksum(token);
-                    const currentBalance = d.tokenBalances[account]?.[chainId]?.[tokenAddress];
+                    const currentBalance = d.tokenBalances[lowerCaseAccount]?.[chainId]?.[tokenAddress];
                     // Only update if the balance has actually changed
                     if (currentBalance !== newBalance) {
-                        ((_c = ((_a = d.tokenBalances)[_b = account] ?? (_a[_b] = {})))[chainId] ?? (_c[chainId] = {}))[tokenAddress] = newBalance;
+                        ((_b = ((_a = d.tokenBalances)[lowerCaseAccount] ?? (_a[lowerCaseAccount] = {})))[chainId] ?? (_b[chainId] = {}))[tokenAddress] = newBalance;
                     }
                 }
             });
diff --git a/dist/TokenBalancesController.mjs b/dist/TokenBalancesController.mjs
index f64d13f8de56631345a44e6ebb025e62e03f51bc..99aa7f27c574c94b26daa56091ac50d15281dd30 100644
--- a/dist/TokenBalancesController.mjs
+++ b/dist/TokenBalancesController.mjs
@@ -531,14 +531,16 @@ export class TokenBalancesController extends StaticIntervalPollingController() {
             }
             // Update with actual fetched balances only if the value has changed
             aggregated.forEach(({ success, value, account, token, chainId }) => {
-                var _a, _b, _c;
+                var _a, _b;
                 if (success && value !== undefined) {
+                    // Ensure all accounts we add/update are in lower-case
+                    const lowerCaseAccount = account.toLowerCase();
                     const newBalance = toHex(value);
                     const tokenAddress = checksum(token);
-                    const currentBalance = d.tokenBalances[account]?.[chainId]?.[tokenAddress];
+                    const currentBalance = d.tokenBalances[lowerCaseAccount]?.[chainId]?.[tokenAddress];
                     // Only update if the balance has actually changed
                     if (currentBalance !== newBalance) {
-                        ((_c = ((_a = d.tokenBalances)[_b = account] ?? (_a[_b] = {})))[chainId] ?? (_c[chainId] = {}))[tokenAddress] = newBalance;
+                        ((_b = ((_a = d.tokenBalances)[lowerCaseAccount] ?? (_a[lowerCaseAccount] = {})))[chainId] ?? (_b[chainId] = {}))[tokenAddress] = newBalance;
                     }
                 }
             });
