diff --git a/dist/KeyringController.js b/dist/KeyringController.js
index d8b1e1b8da74d1647e6fbe50cf94f63bf9fd28cf..6d6d7e590d27be0a9807154dfaac3ce3d1059a75 100644
--- a/dist/KeyringController.js
+++ b/dist/KeyringController.js
@@ -812,7 +812,17 @@ _KeyringController_keyring = new WeakMap(), _KeyringController_qrKeyringStateLis
         }
     });
 }, _KeyringController_fullUpdate = function _KeyringController_fullUpdate() {
-    this.update(() => (Object.assign(Object.assign({}, __classPrivateFieldGet(this, _KeyringController_keyring, "f").store.getState()), __classPrivateFieldGet(this, _KeyringController_keyring, "f").memStore.getState())));
+    const { vault } = __classPrivateFieldGet(this, _KeyringController_keyring, "f").store.getState();
+    const { isUnlocked, keyrings, encryptionKey, encryptionSalt } = 
+        __classPrivateFieldGet(this, _KeyringController_keyring, "f").memStore.getState();
+
+    this.update(() => ({
+        vault,
+        keyrings,
+        isUnlocked,
+        encryptionKey,
+        encryptionSalt,
+    }));
 }, _KeyringController_handleLock = function _KeyringController_handleLock() {
     this.messagingSystem.publish(`${name}:lock`);
 }, _KeyringController_handleUnlock = function _KeyringController_handleUnlock() {
