diff --git a/PATCH.txt b/PATCH.txt
new file mode 100644
index 0000000000000000000000000000000000000000..78b9156dc2b0bf7c33dadf325cb3ec0bfae71ccb
--- /dev/null
+++ b/PATCH.txt
@@ -0,0 +1,3 @@
+We remove `lookupNetwork` from `initializeProvider` in the network controller to prevent network requests before user onboarding is completed.
+The network lookup is done after onboarding is completed, and when the extension reloads if onboarding has been completed.
+This patch is part of a temporary fix that will be reverted soon to make way for a more permanent solution. https://github.com/MetaMask/metamask-extension/pull/23005
diff --git a/dist/chunk-6PT34M4U.mjs b/dist/chunk-6PT34M4U.mjs
index 97bb291b7613f2bf5f5fc021c6d1eb5aac02230e..b81690a756b150bf68d5911a6ce29afea782ea3e 100644
--- a/dist/chunk-6PT34M4U.mjs
+++ b/dist/chunk-6PT34M4U.mjs
@@ -315,7 +315,6 @@ var NetworkController = class extends BaseController {
    */
   async initializeProvider() {
     __privateMethod(this, _applyNetworkSelection, applyNetworkSelection_fn).call(this, this.state.selectedNetworkClientId);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
diff --git a/dist/chunk-WT5ZBF4X.js b/dist/chunk-WT5ZBF4X.js
index 4c79524c78724115e52a92c5d43372e3a24e68c7..a72dff7b7230385314044964c47bd0fde00987bc 100644
--- a/dist/chunk-WT5ZBF4X.js
+++ b/dist/chunk-WT5ZBF4X.js
@@ -315,7 +315,6 @@ var NetworkController = class extends _basecontroller.BaseController {
    */
   async initializeProvider() {
     _chunkZ4BLTVTBjs.__privateMethod.call(void 0, this, _applyNetworkSelection, applyNetworkSelection_fn).call(this, this.state.selectedNetworkClientId);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
