diff --git a/PATCH.txt b/PATCH.txt
new file mode 100644
index 0000000000000000000000000000000000000000..78b9156dc2b0bf7c33dadf325cb3ec0bfae71ccb
--- /dev/null
+++ b/PATCH.txt
@@ -0,0 +1,3 @@
+We remove `lookupNetwork` from `initializeProvider` in the network controller to prevent network requests before user onboarding is completed.
+The network lookup is done after onboarding is completed, and when the extension reloads if onboarding has been completed.
+This patch is part of a temporary fix that will be reverted soon to make way for a more permanent solution. https://github.com/MetaMask/metamask-extension/pull/23005
diff --git a/dist/chunk-X2EGGP54.mjs b/dist/chunk-X2EGGP54.mjs
index 37ea77347ca8c90b0b0d45e74316f4955dcb62d6..dab1a1c74e6d8b0fc5c997e535009bedb4acaf97 100644
--- a/dist/chunk-X2EGGP54.mjs
+++ b/dist/chunk-X2EGGP54.mjs
@@ -459,7 +459,6 @@ var NetworkController = class extends BaseController {
    */
   async initializeProvider() {
     __privateMethod(this, _applyNetworkSelection, applyNetworkSelection_fn).call(this, this.state.selectedNetworkClientId);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
diff --git a/dist/chunk-ZTO4PAJT.js b/dist/chunk-ZTO4PAJT.js
index 5ab61706783e7441ba744f32f79f61b11dd4fadc..fef5dbf4c64a67efde693b64befddb738297fcab 100644
--- a/dist/chunk-ZTO4PAJT.js
+++ b/dist/chunk-ZTO4PAJT.js
@@ -459,7 +459,6 @@ var NetworkController = class extends _basecontroller.BaseController {
    */
   async initializeProvider() {
     _chunkZ4BLTVTBjs.__privateMethod.call(void 0, this, _applyNetworkSelection, applyNetworkSelection_fn).call(this, this.state.selectedNetworkClientId);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
