diff --git a/dist/constants/bridge.cjs b/dist/constants/bridge.cjs
index 4d169e95dc5df859b7cd14235231c8ebf054071e..063e016199a2e6fea244fc2e42c8c3c0cb3d1bc5 100644
--- a/dist/constants/bridge.cjs
+++ b/dist/constants/bridge.cjs
@@ -23,7 +23,6 @@ exports.ALLOWED_BRIDGE_CHAIN_IDS = [
     keyring_api_1.SolScope.Mainnet,
     keyring_api_1.BtcScope.Mainnet,
     keyring_api_1.TrxScope.Mainnet,
-    keyring_api_1.XlmScope.Pubnet,
 ];
 exports.BRIDGE_DEV_API_BASE_URL = 'https://bridge.dev-api.cx.metamask.io';
 exports.BRIDGE_PROD_API_BASE_URL = 'https://bridge.api.cx.metamask.io';
@@ -47,7 +46,6 @@ exports.DEFAULT_CHAIN_RANKING = [
     { chainId: 'bip122:000000000019d6689c085ae165831e93', name: 'BTC' },
     { chainId: 'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp', name: 'Solana' },
     { chainId: 'tron:728126428', name: 'Tron' },
-    { chainId: 'stellar:pubnet', name: 'Stellar' },
     { chainId: 'eip155:8453', name: 'Base' },
     { chainId: 'eip155:42161', name: 'Arbitrum' },
     { chainId: 'eip155:59144', name: 'Linea' },
diff --git a/dist/constants/bridge.d.cts b/dist/constants/bridge.d.cts
index 681da522ddf1e2d06aebb8d6a770e3cd2cd441ce..f42154b6b75db74b7c84e9e3a32bfe7a9f905dbb 100644
--- a/dist/constants/bridge.d.cts
+++ b/dist/constants/bridge.d.cts
@@ -1,7 +1,7 @@
-import { BtcScope, SolScope, TrxScope, XlmScope } from "@metamask/keyring-api";
+import { BtcScope, SolScope, TrxScope } from "@metamask/keyring-api";
 import type { Hex } from "@metamask/utils";
 import type { BridgeControllerState, FeatureFlagsPlatformConfig } from "../types.cjs";
-export declare const ALLOWED_BRIDGE_CHAIN_IDS: readonly ["0x1", "0x38", "0x89", "0x144", "0xa86a", "0xa", "0xa4b1", "0xe708", "0x2105", "0x531", "0x8f", "0x3e7", "0x10e6", "0x13b2", "0x1237", SolScope.Mainnet, BtcScope.Mainnet, TrxScope.Mainnet, XlmScope.Pubnet];
+export declare const ALLOWED_BRIDGE_CHAIN_IDS: readonly ["0x1", "0x38", "0x89", "0x144", "0xa86a", "0xa", "0xa4b1", "0xe708", "0x2105", "0x531", "0x8f", "0x3e7", "0x10e6", "0x13b2", "0x1237", SolScope.Mainnet, BtcScope.Mainnet, TrxScope.Mainnet];
 export type AllowedBridgeChainIds = (typeof ALLOWED_BRIDGE_CHAIN_IDS)[number];
 export declare const BRIDGE_DEV_API_BASE_URL = "https://bridge.dev-api.cx.metamask.io";
 export declare const BRIDGE_PROD_API_BASE_URL = "https://bridge.api.cx.metamask.io";
@@ -34,8 +34,6 @@ export declare const DEFAULT_CHAIN_RANKING: readonly [{
     readonly chainId: "tron:728126428";
     readonly name: "Tron";
 }, {
-    readonly chainId: "stellar:pubnet";
-    readonly name: "Stellar";
 }, {
     readonly chainId: "eip155:8453";
     readonly name: "Base";
diff --git a/dist/constants/bridge.d.mts b/dist/constants/bridge.d.mts
index 81663828b77d36b5bfc015be95eb208de6d61d8e..d356c392e249950bd2874316bf2573b3dba23c07 100644
--- a/dist/constants/bridge.d.mts
+++ b/dist/constants/bridge.d.mts
@@ -1,7 +1,7 @@
-import { BtcScope, SolScope, TrxScope, XlmScope } from "@metamask/keyring-api";
+import { BtcScope, SolScope, TrxScope } from "@metamask/keyring-api";
 import type { Hex } from "@metamask/utils";
 import type { BridgeControllerState, FeatureFlagsPlatformConfig } from "../types.mjs";
-export declare const ALLOWED_BRIDGE_CHAIN_IDS: readonly ["0x1", "0x38", "0x89", "0x144", "0xa86a", "0xa", "0xa4b1", "0xe708", "0x2105", "0x531", "0x8f", "0x3e7", "0x10e6", "0x13b2", "0x1237", SolScope.Mainnet, BtcScope.Mainnet, TrxScope.Mainnet, XlmScope.Pubnet];
+export declare const ALLOWED_BRIDGE_CHAIN_IDS: readonly ["0x1", "0x38", "0x89", "0x144", "0xa86a", "0xa", "0xa4b1", "0xe708", "0x2105", "0x531", "0x8f", "0x3e7", "0x10e6", "0x13b2", "0x1237", SolScope.Mainnet, BtcScope.Mainnet, TrxScope.Mainnet];
 export type AllowedBridgeChainIds = (typeof ALLOWED_BRIDGE_CHAIN_IDS)[number];
 export declare const BRIDGE_DEV_API_BASE_URL = "https://bridge.dev-api.cx.metamask.io";
 export declare const BRIDGE_PROD_API_BASE_URL = "https://bridge.api.cx.metamask.io";
@@ -34,8 +34,6 @@ export declare const DEFAULT_CHAIN_RANKING: readonly [{
     readonly chainId: "tron:728126428";
     readonly name: "Tron";
 }, {
-    readonly chainId: "stellar:pubnet";
-    readonly name: "Stellar";
 }, {
     readonly chainId: "eip155:8453";
     readonly name: "Base";
diff --git a/dist/constants/bridge.mjs b/dist/constants/bridge.mjs
index 01da97226b25daaeee44ff79572cb58e8cf1c788..3987cd0ed09e0ead84bea1c08293637b6352d53f 100644
--- a/dist/constants/bridge.mjs
+++ b/dist/constants/bridge.mjs
@@ -1,5 +1,5 @@
 import { AddressZero } from "@ethersproject/constants";
-import { BtcScope, SolScope, TrxScope, XlmScope } from "@metamask/keyring-api";
+import { BtcScope, SolScope, TrxScope } from "@metamask/keyring-api";
 import { CHAIN_IDS } from "./chains.mjs";
 export const ALLOWED_BRIDGE_CHAIN_IDS = [
     CHAIN_IDS.MAINNET,
@@ -20,7 +20,6 @@ export const ALLOWED_BRIDGE_CHAIN_IDS = [
     SolScope.Mainnet,
     BtcScope.Mainnet,
     TrxScope.Mainnet,
-    XlmScope.Pubnet,
 ];
 export const BRIDGE_DEV_API_BASE_URL = 'https://bridge.dev-api.cx.metamask.io';
 export const BRIDGE_PROD_API_BASE_URL = 'https://bridge.api.cx.metamask.io';
@@ -44,7 +43,6 @@ export const DEFAULT_CHAIN_RANKING = [
     { chainId: 'bip122:000000000019d6689c085ae165831e93', name: 'BTC' },
     { chainId: 'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp', name: 'Solana' },
     { chainId: 'tron:728126428', name: 'Tron' },
-    { chainId: 'stellar:pubnet', name: 'Stellar' },
     { chainId: 'eip155:8453', name: 'Base' },
     { chainId: 'eip155:42161', name: 'Arbitrum' },
     { chainId: 'eip155:59144', name: 'Linea' },
