diff --git a/dist/utils/quote.cjs b/dist/utils/quote.cjs
index f09b7c3e34ca626c93a80cf7c1a2b528d5e07e3f..b8f4dcfbe8ca15a5058d711ec29753893201d5f0 100644
--- a/dist/utils/quote.cjs
+++ b/dist/utils/quote.cjs
@@ -134,10 +134,9 @@ const calcTotalGasFee = ({ approvalGasLimit, resetApprovalGasLimit, tradeGasLimi
     const totalGasLimitInDec = new bignumber_js_1.BigNumber(tradeGasLimit?.toString() ?? '0')
         .plus(approvalGasLimit?.toString() ?? '0')
         .plus(resetApprovalGasLimit?.toString() ?? '0');
-    const totalFeePerGasInDecGwei = new bignumber_js_1.BigNumber(feePerGasInDecGwei).plus(priorityFeePerGasInDecGwei);
     const l1GasFeesInDecGWei = (0, controller_utils_1.weiHexToGweiDec)((0, controller_utils_1.toHex)(l1GasFeesInHexWei ?? '0'));
     const gasFeesInDecGwei = totalGasLimitInDec
-        .times(totalFeePerGasInDecGwei)
+        .times(priorityFeePerGasInDecGwei)
         .plus(l1GasFeesInDecGWei);
     const gasFeesInDecEth = gasFeesInDecGwei.times(new bignumber_js_1.BigNumber(10).pow(-9));
     const gasFeesInDisplayCurrency = nativeToDisplayCurrencyExchangeRate
