diff --git a/dist/chunk-4ZD3DTQ7.js b/dist/chunk-4ZD3DTQ7.js
index e172d15715b3cb4f26c42e51f8c7ff7394075bfe..148c802bac1d79f784618d18fa61d9de3330cae7 100644
--- a/dist/chunk-4ZD3DTQ7.js
+++ b/dist/chunk-4ZD3DTQ7.js
@@ -341,7 +341,6 @@ var NetworkController = class extends _basecontroller.BaseController {
   async initializeProvider() {
     _chunkZ4BLTVTBjs.__privateMethod.call(void 0, this, _ensureAutoManagedNetworkClientRegistryPopulated, ensureAutoManagedNetworkClientRegistryPopulated_fn).call(this);
     _chunkZ4BLTVTBjs.__privateMethod.call(void 0, this, _applyNetworkSelection, applyNetworkSelection_fn).call(this);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
diff --git a/dist/chunk-UG2NYGJD.mjs b/dist/chunk-UG2NYGJD.mjs
index c39eb49a4a1d2b4ddb78aadb4fb03446b1705528..cd7ce41f31434e66b56f584c177d4952e334e212 100644
--- a/dist/chunk-UG2NYGJD.mjs
+++ b/dist/chunk-UG2NYGJD.mjs
@@ -341,7 +341,6 @@ var NetworkController = class extends BaseController {
   async initializeProvider() {
     __privateMethod(this, _ensureAutoManagedNetworkClientRegistryPopulated, ensureAutoManagedNetworkClientRegistryPopulated_fn).call(this);
     __privateMethod(this, _applyNetworkSelection, applyNetworkSelection_fn).call(this);
-    await this.lookupNetwork();
   }
   /**
    * Refreshes the network meta with EIP-1559 support and the network status
