diff --git a/dist/chunk-37VHIRUJ.js b/dist/chunk-37VHIRUJ.js
index a909a4ef20305665a07db5c25b4a9ff7eb0a447e..98dd75bf33a9716dc6cca96a38d184645f6ec033 100644
--- a/dist/chunk-37VHIRUJ.js
+++ b/dist/chunk-37VHIRUJ.js
@@ -53,8 +53,8 @@ function assertIsKeyringOrigins(value, ErrorWrapper) {
 }
 function createOriginRegExp(matcher) {
   const escaped = matcher.replace(/[.*+?^${}()|[\]\\]/gu, "\\$&");
-  const regex = escaped.replace(/\*/gu, ".*");
-  return RegExp(regex, "u");
+  const regex = escaped.replace(/\\\*/gu, '.*');
+  return RegExp(`${regex}$`, 'u');
 }
 function checkAllowedOrigin(matcher, origin) {
   if (matcher === "*" || matcher === origin) {
diff --git a/dist/chunk-K2OTEZZZ.mjs b/dist/chunk-K2OTEZZZ.mjs
index 15be5da7563a5bdf464d7e9c28ed6f04863e378a..7f38bf328e71c1feb2b8850ba050ce9e55801668 100644
--- a/dist/chunk-K2OTEZZZ.mjs
+++ b/dist/chunk-K2OTEZZZ.mjs
@@ -53,8 +53,8 @@ function assertIsKeyringOrigins(value, ErrorWrapper) {
 }
 function createOriginRegExp(matcher) {
   const escaped = matcher.replace(/[.*+?^${}()|[\]\\]/gu, "\\$&");
-  const regex = escaped.replace(/\*/gu, ".*");
-  return RegExp(regex, "u");
+  const regex = escaped.replace(/\\\*/gu, '.*');
+  return RegExp(`${regex}$`, 'u');
 }
 function checkAllowedOrigin(matcher, origin) {
   if (matcher === "*" || matcher === origin) {
