diff --git a/dist/SignatureController.js b/dist/SignatureController.js
index 46d4b4d0553f86d368d30b7e90a9dc2e03d26ef9..e7063a3753bc3821e661c11132e33304b4fce416 100644
--- a/dist/SignatureController.js
+++ b/dist/SignatureController.js
@@ -280,8 +280,11 @@ _SignatureController_isEthSignEnabled = new WeakMap(), _SignatureController_getA
                 resultCallbacks = acceptResult.resultCallbacks;
             }
             catch (_a) {
+                signaturePromise.catch(() => {
+                    // Expecting reject error but throwing manually rather than waiting
+                });
                 __classPrivateFieldGet(this, _SignatureController_instances, "m", _SignatureController_cancelAbstractMessage).call(this, messageManager, messageId);
-                throw eth_rpc_errors_1.ethErrors.provider.userRejectedRequest('User rejected the request.');
+                throw eth_rpc_errors_1.ethErrors.provider.userRejectedRequest(`MetaMask ${messageName} Signature: User denied message signature.`);
             }
             yield signMessage(messageParamsWithId, signingOpts);
             const signatureResult = yield signaturePromise;
