diff --git a/dist/RatesController/RatesController.cjs b/dist/RatesController/RatesController.cjs
index ecf10ece556eaa00f5ea967a8829574b22f29442..deb293761b54a60add7c4f7aa17130ae95187a79 100644
--- a/dist/RatesController/RatesController.cjs
+++ b/dist/RatesController/RatesController.cjs
@@ -84,6 +84,7 @@ class RatesController extends base_controller_1.BaseController {
             return;
         }
         this.messagingSystem.publish(`${exports.name}:pollingStarted`);
+        await __classPrivateFieldGet(this, _RatesController_instances, "m", _RatesController_updateRates).call(this);
         __classPrivateFieldSet(this, _RatesController_intervalId, setInterval(() => {
             __classPrivateFieldGet(this, _RatesController_instances, "m", _RatesController_executePoll).call(this).catch(console.error);
         }, __classPrivateFieldGet(this, _RatesController_intervalLength, "f")), "f");
