diff --git a/dist/NftDetectionController.js b/dist/NftDetectionController.js
index aad63ecf91b2037a40dfb435e0363ab588d22fa3..24373e328d3600d1168914a3dc0bbbd905b19ebe 100644
--- a/dist/NftDetectionController.js
+++ b/dist/NftDetectionController.js
@@ -65,8 +65,6 @@ class NftDetectionController extends polling_controller_1.StaticIntervalPollingC
             if (selectedAddress !== previouslySelectedAddress ||
                 !useNftDetection !== disabled) {
                 this.configure({ selectedAddress, disabled: !useNftDetection });
-            }
-            if (useNftDetection !== undefined) {
                 if (useNftDetection) {
                     this.start();
                 }
