diff --git a/dist/utils/transaction.cjs b/dist/utils/transaction.cjs
index 5ba85f2f336d8cf2bda4d71cf625901251bcb6f2..746b9412f93c82513831d476c49925193e31ec9c 100644
--- a/dist/utils/transaction.cjs
+++ b/dist/utils/transaction.cjs
@@ -238,7 +238,11 @@ const getAddTransactionBatchParams = async ({ messenger, isBridgeTx, approval, r
     // Enable 7702 batching when the quote includes gasless 7702 support,
     // or when the account is already delegated (to avoid the in-flight
     // transaction limit for delegated accounts)
-    const disable7702 = !skipGasFields && !isDelegatedAccount;
+    let disable7702 = !skipGasFields && !isDelegatedAccount;
+    // For gasless transactions with STX/sendBundle we keep disabling 7702.
+    if (gasIncluded && !gasIncluded7702) {
+        disable7702 = true;
+    }
     const transactions = [];
     if (resetApproval) {
         const gasFees = await (0, gas_1.calculateGasFees)(skipGasFields, messenger, estimateGasFeeFn, resetApproval, networkClientId, hexChainId, isGasless ? txFee : undefined);
