diff --git a/dist/NftDetectionController.cjs b/dist/NftDetectionController.cjs
index 8e69510e879c92c45061b6c36b0915850e36da34..b31c73d5d043ab224c9735028bbf24c9d2e185cc 100644
--- a/dist/NftDetectionController.cjs
+++ b/dist/NftDetectionController.cjs
@@ -27,6 +27,7 @@ const supportedNftDetectionNetworks = new Set([
     '0x1', // Mainnet
     '0xe708', // Linea Mainnet
     '0x531', // Sei
+    '0x8f', // Monad
 ]);
 var BlockaidResultType;
 (function (BlockaidResultType) {
diff --git a/dist/token-prices-service/codefi-v2.cjs b/dist/token-prices-service/codefi-v2.cjs
index 34f7bcf4dea1b8d6a1ea45051be09059d9d35353..993fa7b14b918fadf8430ebe9fd525fe7cc6f43a 100644
--- a/dist/token-prices-service/codefi-v2.cjs
+++ b/dist/token-prices-service/codefi-v2.cjs
@@ -98,6 +98,8 @@ exports.SUPPORTED_CURRENCIES = [
     'mxn',
     // Malaysian Ringgit
     'myr',
+    // Monad,
+    'mon',
     // Nigerian Naira
     'ngn',
     // Norwegian Krone
