diff --git a/dist/constants/tokens.cjs b/dist/constants/tokens.cjs
index db43ed22aead16bea7736dbd1917cecd29b38860..1e78ff1a2dee8e12563f73f2b65fea0d83952f94 100644
--- a/dist/constants/tokens.cjs
+++ b/dist/constants/tokens.cjs
@@ -4,7 +4,6 @@ exports.SYMBOL_TO_SLIP44_MAP = exports.SWAPS_CHAINID_DEFAULT_TOKEN_MAP = void 0;
 const keyring_api_1 = require("@metamask/keyring-api");
 const chains_1 = require("./chains.cjs");
 const DEFAULT_TOKEN_ADDRESS = '0x0000000000000000000000000000000000000000';
-const DEFAULT_SOLANA_TOKEN_ADDRESS = `${keyring_api_1.SolScope.Mainnet}/slip44:501`;
 const CURRENCY_SYMBOLS = {
     ARBITRUM: 'ETH',
     AVALANCHE: 'AVAX',
@@ -97,7 +96,7 @@ const BASE_SWAPS_TOKEN_OBJECT = {
 const SOLANA_SWAPS_TOKEN_OBJECT = {
     symbol: CURRENCY_SYMBOLS.SOL,
     name: 'Solana',
-    address: DEFAULT_SOLANA_TOKEN_ADDRESS,
+    address: DEFAULT_TOKEN_ADDRESS,
     decimals: 9,
     iconUrl: '',
 };
diff --git a/dist/constants/tokens.mjs b/dist/constants/tokens.mjs
index 27280664758ee3052790aa5c6d54296a010fbbd3..ee0d8e7cbc27d72726111971158921e930504d6b 100644
--- a/dist/constants/tokens.mjs
+++ b/dist/constants/tokens.mjs
@@ -1,7 +1,6 @@
 import { SolScope } from "@metamask/keyring-api";
 import { CHAIN_IDS } from "./chains.mjs";
 const DEFAULT_TOKEN_ADDRESS = '0x0000000000000000000000000000000000000000';
-const DEFAULT_SOLANA_TOKEN_ADDRESS = `${SolScope.Mainnet}/slip44:501`;
 const CURRENCY_SYMBOLS = {
     ARBITRUM: 'ETH',
     AVALANCHE: 'AVAX',
@@ -94,7 +93,7 @@ const BASE_SWAPS_TOKEN_OBJECT = {
 const SOLANA_SWAPS_TOKEN_OBJECT = {
     symbol: CURRENCY_SYMBOLS.SOL,
     name: 'Solana',
-    address: DEFAULT_SOLANA_TOKEN_ADDRESS,
+    address: DEFAULT_TOKEN_ADDRESS,
     decimals: 9,
     iconUrl: '',
 };
