diff --git a/dist/mnemonic.mjs b/dist/mnemonic.mjs
index 76818e74131e8b826f4da2f18c0105aef67f76af..219980e0708310173ada8f6646d7abeb69f06ae2 100644
--- a/dist/mnemonic.mjs
+++ b/dist/mnemonic.mjs
@@ -1,5 +1,4 @@
-import $metamaskscurebipdistwordlistsenglishjs from "@metamask/scure-bip39/dist/wordlists/english.js";
-const { wordlist } = $metamaskscurebipdistwordlistsenglishjs;
+import { wordlist } from "@metamask/scure-bip39/dist/wordlists/english.js";
 /**
  * Transform a typed array containing mnemonic data to the seed phrase.
  * @param uint8Array - Typed array containing mnemonic data.
