diff --git a/dist/utils.js b/dist/utils.js
index 0a77dd709e786ee586b70677fa951f34d5097e34..24dbc12c31686a3ad1b78de702801e19a239f0fc 100644
--- a/dist/utils.js
+++ b/dist/utils.js
@@ -12,11 +12,11 @@ const uuid_1 = require("uuid");
  * @returns The name of the keyring type.
  */
 function keyringTypeToName(keyringType) {
-    // Custody keyrings are a special case, as they are not a single type
-    // they just start with the prefix `Custody`
-    if ((0, keyring_controller_1.isCustodyKeyring)(keyringType)) {
+    // Custody Keyrings follow this pattern
+    if (/^Custody.*/.test(keyringType)) {
         return 'Custody';
     }
+
     switch (keyringType) {
         case keyring_controller_1.KeyringTypes.simple: {
             return 'Account';
@@ -39,6 +39,9 @@ function keyringTypeToName(keyringType) {
         case keyring_controller_1.KeyringTypes.snap: {
             return 'Snap Account';
         }
+        case keyring_controller_1.KeyringTypes.custody: {
+            return 'Custody';
+        }
         default: {
             throw new Error(`Unknown keyring ${keyringType}`);
         }
