diff --git a/dist/KeyringController.js b/dist/KeyringController.js
index da9523afe0a83e3fa298b47a518cb583eb509e52..8fe701103975335ff02ed8195f250f35eedccbab 100644
--- a/dist/KeyringController.js
+++ b/dist/KeyringController.js
@@ -662,7 +662,6 @@ class KeyringController extends base_controller_1.BaseControllerV2 {
                     throw new Error('Seed phrase imported different accounts.');
                 }
             });
-            return seedWords;
         });
     }
     // QR Hardware related methods
