diff --git a/dist/NetworkController.cjs b/dist/NetworkController.cjs
index e0bf9fc4b29ba759fb9efd736d7c88cd5c14becd..d657d3905369e05281d416166e223027d5b10981 100644
--- a/dist/NetworkController.cjs
+++ b/dist/NetworkController.cjs
@@ -456,7 +456,6 @@ class NetworkController extends base_controller_1.BaseController {
      */
     async initializeProvider() {
         __classPrivateFieldGet(this, _NetworkController_instances, "m", _NetworkController_applyNetworkSelection).call(this, this.state.selectedNetworkClientId);
-        await this.lookupNetwork();
     }
     /**
      * Refreshes the network meta with EIP-1559 support and the network status
diff --git a/dist/NetworkController.mjs b/dist/NetworkController.mjs
index 8cadd1439d0dd3aa77d3f5141fdb2d9a7115b10b..6681eb00765127e21a7deb08fe5166ba063aa0dc 100644
--- a/dist/NetworkController.mjs
+++ b/dist/NetworkController.mjs
@@ -432,7 +432,6 @@ export class NetworkController extends BaseController {
      */
     async initializeProvider() {
         __classPrivateFieldGet(this, _NetworkController_instances, "m", _NetworkController_applyNetworkSelection).call(this, this.state.selectedNetworkClientId);
-        await this.lookupNetwork();
     }
     /**
      * Refreshes the network meta with EIP-1559 support and the network status
diff --git a/dist/PATCH.txt b/dist/PATCH.txt
new file mode 100644
index 0000000000000000000000000000000000000000..376255036ca54b83a3f3c3e0277c2666473df30e
--- /dev/null
+++ b/dist/PATCH.txt
@@ -0,0 +1,4 @@
+We remove lookupNetwork from initializeProvider in the network controller to prevent network requests before user onboarding is completed.
+The network lookup is done after onboarding is completed, and when the extension reloads if onboarding has been completed.
+This patch is part of a temporary fix that will be reverted soon to make way for a more permanent solution. https://github.com/MetaMask/metamask-extension/pull/23005
+You can see the changes before compilation on this branch: https://github.com/MetaMask/core/compare/pnf/ext-23622-review?expand=1
