diff --git a/dist/json-rpc.cjs b/dist/json-rpc.cjs
index 6061f7b8b42f0521b0718d616e5a12a1a7520068..11d0233a7bd4b610a99da6a3d105840e88e108e6 100644
--- a/dist/json-rpc.cjs
+++ b/dist/json-rpc.cjs
@@ -68,7 +68,7 @@ function createOriginRegExp(matcher) {
     const escaped = matcher.replace(/[.*+?^${}()|[\]\\]/gu, '\\$&');
     // Support wildcards
     const regex = escaped.replace(/\\\*/gu, '.*');
-    return RegExp(`${regex}$`, 'u');
+    return RegExp(`^${regex}$`, 'u');
 }
 /**
  * Check whether an origin is allowed or not using a matcher string.
diff --git a/dist/json-rpc.mjs b/dist/json-rpc.mjs
index bfa1c2dbbed46a2221ef708afdb97b15db84bc1b..81bc2150cf5d6a9bdabe8d43b04352b299bc1c4d 100644
--- a/dist/json-rpc.mjs
+++ b/dist/json-rpc.mjs
@@ -63,7 +63,7 @@ function createOriginRegExp(matcher) {
     const escaped = matcher.replace(/[.*+?^${}()|[\]\\]/gu, '\\$&');
     // Support wildcards
     const regex = escaped.replace(/\\\*/gu, '.*');
-    return RegExp(`${regex}$`, 'u');
+    return RegExp(`^${regex}$`, 'u');
 }
 /**
  * Check whether an origin is allowed or not using a matcher string.
