diff --git a/dist/ledger-keyring.cjs b/dist/ledger-keyring.cjs
index b8bdf04a3de250bcdbe1064382682bc18f460cbc..9d93bdfe7dab04aae662a30e4f2342441d5a2261 100644
--- a/dist/ledger-keyring.cjs
+++ b/dist/ledger-keyring.cjs
@@ -277,6 +277,12 @@ class LedgerKeyring {
             });
         }
         catch (error) {
+            /********** 
+            * Patch for error handling for Device Management kit 
+            **********/
+            if(error?.errorCode === "6985" && error?.message === 'Condition not satisfied'){
+                throw new Error('Ledger: User rejected the transaction');
+            }
             throw error instanceof Error
                 ? error
                 : new Error('Ledger: Unknown error while signing message');
@@ -343,6 +349,15 @@ class LedgerKeyring {
             });
         }
         catch (error) {
+            /********** 
+            * Patch for error handling for Device Management kit 
+            **********/
+            if(error?.errorCode === "6985" && error?.message === 'Condition not satisfied'){
+                throw new Error('Ledger: User rejected the transaction');
+            }
+            if(error?.errorCode === "6a80" && error?.message === 'Invalid data'){
+               throw new Error('Ledger: Blind signing not enabled');
+            }
             throw error instanceof Error
                 ? error
                 : new Error('Ledger: Unknown error while signing message');
@@ -409,6 +424,15 @@ _LedgerKeyring_instances = new WeakSet(), _LedgerKeyring_migrateAccountDetails =
         });
     }
     catch (error) {
+        /********** 
+        * Patch for error handling for Device Management kit 
+        **********/
+        if(error?.errorCode === "6985" && error?.message === 'Condition not satisfied'){
+            throw new Error('Ledger: User rejected the transaction');
+        }
+        if(error?.errorCode === "6a80" && error?.message === 'Invalid data'){
+            throw new Error('Ledger: Blind signing not enabled');
+        }
         throw error instanceof Error
             ? error
             : new Error('Ledger: Unknown error while signing transaction');
