diff --git a/dist/KeyringController.js b/dist/KeyringController.js
index fc649ea6fc97b905d811b236de638172fb10b548..a856abf0b4942585e0b9b5682b1974bff6f68b1f 100644
--- a/dist/KeyringController.js
+++ b/dist/KeyringController.js
@@ -945,7 +945,6 @@ class KeyringController extends base_controller_1.BaseController {
                     throw new Error('Seed phrase imported different accounts.');
                 }
             });
-            return seedWords;
         });
     }
     // QR Hardware related methods
