diff --git a/dist/chunk-FMZML3V5.js b/dist/chunk-FMZML3V5.js
index 0d1907f91ca9a118498226bbc0c94734af9f6002..7531c49601a13aa132689559ae2c72fb6d18eb09 100644
--- a/dist/chunk-FMZML3V5.js
+++ b/dist/chunk-FMZML3V5.js
@@ -104,20 +104,25 @@ var NftDetectionController = class extends _pollingcontroller.StaticIntervalPoll
   ) {
     // Convert hex chainId to number
     const convertedChainId = (0, _controllerutils.convertHexToDecimal)(chainId).toString();
-    const nftApiResponse = await _controllerutils.fetchWithErrorHandling.call(void 0, {
-      url: this.getOwnerNftApi({
-        chainId: convertedChainId,
-        address,
-        next: cursor,
-      }),
-      options: {
+    const url = this.getOwnerNftApi({
+      chainId: convertedChainId,
+      address,
+      next: cursor,
+    });
+
+    try{
+      const nftApiResponse = await _controllerutils.handleFetch.call(void 0, url,
+      {
         headers: {
           Version: "1"
         },
-      },
-      timeout: 15e3
-    });
-    return nftApiResponse;
+      }
+      );
+      return nftApiResponse;
+    }catch(err){
+      this.updateNftFetchingProgressStatus(false)
+      throw err
+    }
   }
   async _executePoll(networkClientId, options) {
     await this.detectNfts({ networkClientId, userAddress: options.address });
