diff --git a/dist/constants.cjs b/dist/constants.cjs
index 0ce368f00d98bcb38f43e9c7fb608b51b955e7c5..e80711fbd7652b2ced819162f63f3d3354260cec 100644
--- a/dist/constants.cjs
+++ b/dist/constants.cjs
@@ -17,5 +17,6 @@ exports.POPULAR_NETWORKS = [
     '0x8f', // Monad (143)
     '0x10e6', // MegaETH (4326)
     '0x13b2', // Arc (5042)
+    '0x1237', // Robinhood Chain (4663)
 ];
 //# sourceMappingURL=constants.cjs.map
\ No newline at end of file
diff --git a/dist/constants.mjs b/dist/constants.mjs
index 476d382fbfa0cc052a66aae03a58473af8c9f118..96b817adb8bccdfbba3f9f3fca7aa67325ef4df2 100644
--- a/dist/constants.mjs
+++ b/dist/constants.mjs
@@ -14,5 +14,6 @@ export const POPULAR_NETWORKS = [
     '0x8f', // Monad (143)
     '0x10e6', // MegaETH (4326)
     '0x13b2', // Arc (5042)
+    '0x1237', // Robinhood Chain (4663)
 ];
 //# sourceMappingURL=constants.mjs.map
\ No newline at end of file
