diff --git a/dist/utils/validation.cjs b/dist/utils/validation.cjs
index eb116f75643a6607af10eb1669aaabfb48159826..349937fca246b26deac4a428fb3ad93dc7e18a6e 100644
--- a/dist/utils/validation.cjs
+++ b/dist/utils/validation.cjs
@@ -150,7 +150,9 @@ function validateAddress(address, propertyName) {
 function validateVerifyingContract({ data, internalAccounts, origin, }) {
     const verifyingContract = data?.domain?.verifyingContract;
     const isExternal = origin && origin !== approval_controller_1.ORIGIN_METAMASK;
-    if (isExternal &&
+    if (verifyingContract &&
+        typeof verifyingContract === 'string' &&
+        isExternal &&
         internalAccounts.some((internalAccount) => internalAccount.toLowerCase() === verifyingContract.toLowerCase())) {
         throw new Error(`External signature requests cannot use internal accounts as the verifying contract.`);
     }
diff --git a/dist/utils/validation.mjs b/dist/utils/validation.mjs
index 58b55c134527537c0ef1f44876afcf2939e9ed53..d4af6e8a1e8fff08e422ebfa6b9f444dc24c8632 100644
--- a/dist/utils/validation.mjs
+++ b/dist/utils/validation.mjs
@@ -145,7 +145,9 @@ function validateAddress(address, propertyName) {
 function validateVerifyingContract({ data, internalAccounts, origin, }) {
     const verifyingContract = data?.domain?.verifyingContract;
     const isExternal = origin && origin !== ORIGIN_METAMASK;
-    if (isExternal &&
+    if (verifyingContract &&
+        typeof verifyingContract === 'string' &&
+        isExternal &&
         internalAccounts.some((internalAccount) => internalAccount.toLowerCase() === verifyingContract.toLowerCase())) {
         throw new Error(`External signature requests cannot use internal accounts as the verifying contract.`);
     }
