diff --git a/lib/esm/fetch.js b/lib/esm/fetch.js
index 175b639f3da6754f01641fbf74df791b36520c4d..7177380d48fde995fd4f93cff91511e5ec105ed5 100644
--- a/lib/esm/fetch.js
+++ b/lib/esm/fetch.js
@@ -37,9 +37,9 @@ export function fetchEventSource(input, _a) {
             window.clearTimeout(retryTimer);
             curRequestController.abort();
         }
-        inputSignal === null || inputSignal === void 0 ? void 0 : inputSignal.addEventListener('abort', () => {
+        inputSignal === null || inputSignal === void 0 ? void 0 : inputSignal.addEventListener('abort', (e) => {
             dispose();
-            resolve();
+            reject(Object.assign(new Error('Aborted'), { name: 'AbortError', error: e.target.reason }));
         });
         const fetch = inputFetch !== null && inputFetch !== void 0 ? inputFetch : window.fetch;
         const onopen = inputOnOpen !== null && inputOnOpen !== void 0 ? inputOnOpen : defaultOnOpen;
