diff --git a/build/cjs/instrument/fetch.js b/build/cjs/instrument/fetch.js
index 975cd5f0427a9aca809173b28bafb187442b777c..affce36b67d501d5c685df17c481630a527a3bf8 100644
--- a/build/cjs/instrument/fetch.js
+++ b/build/cjs/instrument/fetch.js
@@ -119,7 +119,8 @@ function instrumentFetch(onFetchResolved, skipNativeFetchCheck = false) {

           if (
             shouldEnhance &&
-            error instanceof TypeError &&
+            is.isError(error) &&
+            error.name === 'TypeError' &&
             (error.message === 'Failed to fetch' ||
               error.message === 'Load failed' ||
               error.message === 'NetworkError when attempting to fetch resource.')
diff --git a/build/cjs/utils/aggregate-errors.js b/build/cjs/utils/aggregate-errors.js
index e520588efa588ccb6b886e6054c8acfd82097a6f..765a52c66476c85a391d5c72b0f3d3040ed297b6 100644
--- a/build/cjs/utils/aggregate-errors.js
+++ b/build/cjs/utils/aggregate-errors.js
@@ -13,7 +13,7 @@ function applyAggregateErrorsToEvent(
   event,
   hint,
 ) {
-  if (!event.exception?.values || !hint || !is.isInstanceOf(hint.originalException, Error)) {
+  if (!event.exception?.values || !hint || !is.isError(hint.originalException)) {
     return;
   }

@@ -53,7 +53,7 @@ function aggregateExceptionsFromError(
   let newExceptions = [...prevExceptions];

   // Recursively call this function in order to walk down a chain of errors
-  if (is.isInstanceOf(error[key], Error)) {
+  if (is.isError(error[key])) {
     applyExceptionGroupFieldsForParentException(exception, exceptionId, error);
     const newException = exceptionFromErrorImplementation(parser, error[key] );
     const newExceptionId = newExceptions.length;
@@ -74,7 +74,7 @@ function aggregateExceptionsFromError(
   // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError
   if (isExceptionGroup(error)) {
     error.errors.forEach((childError, i) => {
-      if (is.isInstanceOf(childError, Error)) {
+      if (is.isError(childError)) {
         applyExceptionGroupFieldsForParentException(exception, exceptionId, error);
         const newException = exceptionFromErrorImplementation(parser, childError );
         const newExceptionId = newExceptions.length;
diff --git a/build/cjs/utils/eventbuilder.js b/build/cjs/utils/eventbuilder.js
index 8d8bdaf3aba09d91e5b257c51c62dcc3c232ec2d..cd97cca19202add863f904e18aba7d43d04d7563 100644
--- a/build/cjs/utils/eventbuilder.js
+++ b/build/cjs/utils/eventbuilder.js
@@ -56,7 +56,7 @@ function getErrorPropertyFromObject(obj) {
   for (const prop in obj) {
     if (Object.prototype.hasOwnProperty.call(obj, prop)) {
       const value = obj[prop];
-      if (value instanceof Error) {
+      if (is.isError(value)) {
         return value;
       }
     }
diff --git a/build/esm/instrument/fetch.js b/build/esm/instrument/fetch.js
index a42c87e5df0eac35b3ad4ca2352f485ef87a8e2d..2bdc2950dd9db326fd54aa568482d06775498351 100644
--- a/build/esm/instrument/fetch.js
+++ b/build/esm/instrument/fetch.js
@@ -117,7 +117,8 @@ function instrumentFetch(onFetchResolved, skipNativeFetchCheck = false) {

           if (
             shouldEnhance &&
-            error instanceof TypeError &&
+            isError(error) &&
+            error.name === 'TypeError' &&
             (error.message === 'Failed to fetch' ||
               error.message === 'Load failed' ||
               error.message === 'NetworkError when attempting to fetch resource.')
diff --git a/build/esm/utils/aggregate-errors.js b/build/esm/utils/aggregate-errors.js
index 15ce65f56ca8f4118102949308aa0250413889a7..be49490e40def5d0b4d4dd20ae0966bd97f184b4 100644
--- a/build/esm/utils/aggregate-errors.js
+++ b/build/esm/utils/aggregate-errors.js
@@ -1,4 +1,4 @@
-import { isInstanceOf } from './is.js';
+import { isError } from './is.js';

 /**
  * Creates exceptions inside `event.exception.values` for errors that are nested on properties based on the `key` parameter.
@@ -11,7 +11,7 @@ function applyAggregateErrorsToEvent(
   event,
   hint,
 ) {
-  if (!event.exception?.values || !hint || !isInstanceOf(hint.originalException, Error)) {
+  if (!event.exception?.values || !hint || !isError(hint.originalException)) {
     return;
   }

@@ -51,7 +51,7 @@ function aggregateExceptionsFromError(
   let newExceptions = [...prevExceptions];

   // Recursively call this function in order to walk down a chain of errors
-  if (isInstanceOf(error[key], Error)) {
+  if (isError(error[key])) {
     applyExceptionGroupFieldsForParentException(exception, exceptionId, error);
     const newException = exceptionFromErrorImplementation(parser, error[key] );
     const newExceptionId = newExceptions.length;
@@ -72,7 +72,7 @@ function aggregateExceptionsFromError(
   // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError
   if (isExceptionGroup(error)) {
     error.errors.forEach((childError, i) => {
-      if (isInstanceOf(childError, Error)) {
+      if (isError(childError)) {
         applyExceptionGroupFieldsForParentException(exception, exceptionId, error);
         const newException = exceptionFromErrorImplementation(parser, childError );
         const newExceptionId = newExceptions.length;
diff --git a/build/esm/utils/eventbuilder.js b/build/esm/utils/eventbuilder.js
index 711f343349ae77e3bcfb3d35c1c3d578bfdfa023..236fa1cea5078d46dc6afe4ade24cbaee19fbc93 100644
--- a/build/esm/utils/eventbuilder.js
+++ b/build/esm/utils/eventbuilder.js
@@ -54,7 +54,7 @@ function getErrorPropertyFromObject(obj) {
   for (const prop in obj) {
     if (Object.prototype.hasOwnProperty.call(obj, prop)) {
       const value = obj[prop];
-      if (value instanceof Error) {
+      if (isError(value)) {
         return value;
       }
     }
