diff --git a/lib/index.js b/lib/index.js
index 82a21b70dc18597fdbb1c5bfdbebb10f615c723d..04edf4b3a8a5a2bebda8bf907225b502ab5c30ad 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -109,7 +109,9 @@ const init = async (settings = {}) => {
     _log.enabled = !!_settings.debug;
     window.addEventListener('message', handleMessage);
     window.addEventListener('unload', dispose);
-    await iframe.init(_settings);
+    var modifiedSettings = Object.assign({}, _settings);
+    modifiedSettings.env = 'webextension';
+    await iframe.init(modifiedSettings);
     if (_settings.sharedLogger !== false) {
         iframe.initIframeLogger();
     }
@@ -125,7 +127,9 @@ const call = async (params) => {
         }
         _popupManager.request();
         try {
-            await init(_settings);
+            var modifiedSettings = Object.assign({}, _settings);
+            modifiedSettings.env = 'webextension';
+            await init(modifiedSettings);
         }
         catch (error) {
             if (_popupManager) {
diff --git a/lib/popup/index.js b/lib/popup/index.js
index 6948bdb73b381bb72fb8b87fe2006d0046b65acb..d8f80f77728a18a851da3f03a5ed956cb46fe8e6 100644
--- a/lib/popup/index.js
+++ b/lib/popup/index.js
@@ -272,9 +272,11 @@ class PopupManager extends events_1.default {
                 this.popupPromise.resolve();
             }
             (_b = this.iframeHandshakePromise) === null || _b === void 0 ? void 0 : _b.promise.then(payload => {
+                var modifiedSettings = Object.assign({}, this.settings);
+                modifiedSettings.env = 'webextension';
                 this.channel.postMessage({
                     type: events_2.POPUP.INIT,
-                    payload: Object.assign(Object.assign({}, payload), { settings: this.settings }),
+                    payload: Object.assign(Object.assign({}, payload), { settings: modifiedSettings }),
                 });
             });
         }
