diff --git a/lib/extendStringPrototype.js b/lib/extendStringPrototype.js
index 46fd386a915a67d53fa8c3beefdf74d6c0ed03bc..c7d0fc50f42603463eb8237e2428802ab8831eb9 100644
--- a/lib/extendStringPrototype.js
+++ b/lib/extendStringPrototype.js
@@ -5,7 +5,8 @@ module['exports'] = function() {
   // Extends prototype of native string object to allow for "foo".red syntax
   //
   var addProperty = function(color, func) {
-    String.prototype.__defineGetter__(color, func);
+    // remove prototype mutation so this plays well with LavaMoat
+    // String.prototype.__defineGetter__(color, func);
   };
 
   addProperty('strip', function() {
