diff --git a/dist/purify.cjs.js b/dist/purify.cjs.js
index 0b3b5ea4cce01023f7d6f25874cc97cf299d68c1..3c2ffcc55298b6451e13aa2ec9e15420ec386a5b 100644
--- a/dist/purify.cjs.js
+++ b/dist/purify.cjs.js
@@ -1980,7 +1980,8 @@ function createDOMPurify() {
       the user has requested a DOM object rather than a string */
     IS_EMPTY_INPUT = !dirty;
     if (IS_EMPTY_INPUT) {
-      dirty = '<!-->';
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      dirty = '<!' + '--' + '>';
     }
     /* Stringify, in case dirty is an object */
     if (typeof dirty !== 'string' && !_isNode(dirty)) {
@@ -2046,7 +2047,8 @@ function createDOMPurify() {
     } else if (_isNode(dirty)) {
       /* If dirty is a DOM element, append to an empty document to avoid
          elements being stripped by the parser */
-      body = _initDocument('<!---->');
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      body = _initDocument('<!' + '--' + '--' + '>');
       importedNode = body.ownerDocument.importNode(dirty, true);
       if (importedNode.nodeType === NODE_TYPE.element && importedNode.nodeName === 'BODY') {
         /* Node is already a body, use as is */
diff --git a/dist/purify.es.mjs b/dist/purify.es.mjs
index 10f7e9e2da539632805ad64c3061a2a12dfe36f0..5554bcf4a7819ef8208c8b9d21572e72ca4ce6f2 100644
--- a/dist/purify.es.mjs
+++ b/dist/purify.es.mjs
@@ -1980,7 +1980,8 @@ function createDOMPurify() {
       the user has requested a DOM object rather than a string */
     IS_EMPTY_INPUT = !dirty;
     if (IS_EMPTY_INPUT) {
-      dirty = '<!-->';
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      dirty = '<!' + '--' + '>';
     }
     /* Stringify, in case dirty is an object */
     if (typeof dirty !== 'string' && !_isNode(dirty)) {
@@ -2046,7 +2047,8 @@ function createDOMPurify() {
     } else if (_isNode(dirty)) {
       /* If dirty is a DOM element, append to an empty document to avoid
          elements being stripped by the parser */
-      body = _initDocument('<!---->');
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      body = _initDocument('<!' + '--' + '--' + '>');
       importedNode = body.ownerDocument.importNode(dirty, true);
       if (importedNode.nodeType === NODE_TYPE.element && importedNode.nodeName === 'BODY') {
         /* Node is already a body, use as is */
diff --git a/dist/purify.js b/dist/purify.js
index 2764b60bead93927c3476c83c6d351ba72e80ae7..2d23f4f75d8f3f6dcfe754d854de14adf79caf41 100644
--- a/dist/purify.js
+++ b/dist/purify.js
@@ -1986,7 +1986,8 @@
         the user has requested a DOM object rather than a string */
       IS_EMPTY_INPUT = !dirty;
       if (IS_EMPTY_INPUT) {
-        dirty = '<!-->';
+        // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+        dirty = '<!' + '--' + '>';
       }
       /* Stringify, in case dirty is an object */
       if (typeof dirty !== 'string' && !_isNode(dirty)) {
@@ -2052,7 +2053,8 @@
       } else if (_isNode(dirty)) {
         /* If dirty is a DOM element, append to an empty document to avoid
            elements being stripped by the parser */
-        body = _initDocument('<!---->');
+        // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+        body = _initDocument('<!' + '--' + '--' + '>');
         importedNode = body.ownerDocument.importNode(dirty, true);
         if (importedNode.nodeType === NODE_TYPE.element && importedNode.nodeName === 'BODY') {
           /* Node is already a body, use as is */