diff --git a/dist/purify.cjs.js b/dist/purify.cjs.js
index 91b993d668dd9780a88ea26ba6bf60eb17ef07bb..603ec9c2233ab857afee32fbf4af4c35895c154d 100644
--- a/dist/purify.cjs.js
+++ b/dist/purify.cjs.js
@@ -2128,7 +2128,8 @@ function createDOMPurify() {
       the user has requested a DOM object rather than a string */
     IS_EMPTY_INPUT = !dirty;
     if (IS_EMPTY_INPUT) {
-      dirty = '<!-->';
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      dirty = '<!' + '--' + '>';
     }
     /* Stringify, in case dirty is an object */
     if (typeof dirty !== 'string' && !_isNode(dirty)) {
@@ -2228,7 +2229,8 @@ function createDOMPurify() {
     } else if (_isNode(dirty)) {
       /* If dirty is a DOM element, append to an empty document to avoid
          elements being stripped by the parser */
-      body = _initDocument('<!---->');
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      body = _initDocument('<!' + '--' + '--' + '>');
       importedNode = body.ownerDocument.importNode(dirty, true);
       if (importedNode.nodeType === NODE_TYPE.element && importedNode.nodeName === 'BODY') {
         /* Node is already a body, use as is */
diff --git a/dist/purify.es.mjs b/dist/purify.es.mjs
index 3c361d7afae9348d7e12d046a598a4dabb4ba423..489289ac87b83c35f3c7981d342a71da74f5126c 100644
--- a/dist/purify.es.mjs
+++ b/dist/purify.es.mjs
@@ -2126,7 +2126,8 @@ function createDOMPurify() {
       the user has requested a DOM object rather than a string */
     IS_EMPTY_INPUT = !dirty;
     if (IS_EMPTY_INPUT) {
-      dirty = '<!-->';
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      dirty = '<!' + '--' + '>';
     }
     /* Stringify, in case dirty is an object */
     if (typeof dirty !== 'string' && !_isNode(dirty)) {
@@ -2226,7 +2227,8 @@ function createDOMPurify() {
     } else if (_isNode(dirty)) {
       /* If dirty is a DOM element, append to an empty document to avoid
          elements being stripped by the parser */
-      body = _initDocument('<!---->');
+      // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+      body = _initDocument('<!' + '--' + '--' + '>');
       importedNode = body.ownerDocument.importNode(dirty, true);
       if (importedNode.nodeType === NODE_TYPE.element && importedNode.nodeName === 'BODY') {
         /* Node is already a body, use as is */
diff --git a/dist/purify.js b/dist/purify.js
index 0ec2e6e0158b8b535d7ab50d66104f56de748e22..e9f021c6861d2883fbb38d4953be674a8e136f6a 100644
--- a/dist/purify.js
+++ b/dist/purify.js
@@ -2132,7 +2132,8 @@
         the user has requested a DOM object rather than a string */
       IS_EMPTY_INPUT = !dirty;
       if (IS_EMPTY_INPUT) {
-        dirty = '<!-->';
+        // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+        dirty = '<!' + '--' + '>';
       }
       /* Stringify, in case dirty is an object */
       if (typeof dirty !== 'string' && !_isNode(dirty)) {
@@ -2232,7 +2233,8 @@
       } else if (_isNode(dirty)) {
         /* If dirty is a DOM element, append to an empty document to avoid
            elements being stripped by the parser */
-        body = _initDocument('<!---->');
+        // Modifying to avoid lavamoat SES_HTML_COMMENT_REJECTED
+        body = _initDocument('<!' + '--' + '--' + '>');
         importedNode = body.ownerDocument.importNode(dirty, true);
         if (importedNode.nodeType === NODE_TYPE.element && importedNode.nodeName === 'BODY') {
           /* Node is already a body, use as is */
