diff --git a/index.js b/index.js
index f210ce203391f4a875cfcd4b3be6817d418df5bd..f63621066e35541c05d6975ee259cc9e37d979cb 100644
--- a/index.js
+++ b/index.js
@@ -239,6 +239,11 @@ class KeyringController extends EventEmitter {
   async addNewKeyring(type, opts) {
     const Keyring = this.getKeyringClassForType(type);
     const keyring = new Keyring(opts);
+
+    if(keyring.init){
+      await keyring.init();
+    }
+
     if ((!opts || !opts.mnemonic) && type === KEYRINGS_TYPE_MAP.HD_KEYRING) {
       keyring.generateRandomMnemonic();
       keyring.addAccounts();
@@ -429,7 +434,9 @@ class KeyringController extends EventEmitter {
   async signPersonalMessage(msgParams, opts = {}) {
     const address = normalizeAddress(msgParams.from);
     const keyring = await this.getKeyringForAccount(address);
-    return await keyring.signPersonalMessage(address, msgParams.data, opts);
+    const msgBuffer = await keyring.signPersonalMessage(address, msgParams.data, opts);
+
+    return msgBuffer;
   }
 
   /**
@@ -706,6 +713,11 @@ class KeyringController extends EventEmitter {
       return undefined;
     }
     const keyring = new Keyring();
+
+    if(keyring.init) {
+      await keyring.init();
+    }
+    
     await keyring.deserialize(data);
     // getAccounts also validates the accounts for some keyrings
     await keyring.getAccounts();
