diff --git a/lib/jsdom/browser/Window.js b/lib/jsdom/browser/Window.js
index 40f4ac9498565c67ad916e7a7e2c868e898756d9..17e8e861fd445733f2ed239e0deab51fd10701dc 100644
--- a/lib/jsdom/browser/Window.js
+++ b/lib/jsdom/browser/Window.js
@@ -430,6 +430,14 @@ function Window(options) {
 
       return this._localStorage;
     },
+    set localStorage(value) {
+      Object.defineProperty(this, "localStorage", {
+        value,
+        writable: false,
+        enumerable: true,
+        configurable: true
+      });
+    },
     get sessionStorage() {
       if (idlUtils.implForWrapper(this._document)._origin === "null") {
         throw DOMException.create(window, [
