diff --git a/src/kernelCoreTemplate.js b/src/kernelCoreTemplate.js
index 255b2637167de9e5da435fe2f7ea364cfb22f86b..c229f5067a64d0a2705fa2f07085b6cac42bca9c 100644
--- a/src/kernelCoreTemplate.js
+++ b/src/kernelCoreTemplate.js
@@ -124,7 +124,8 @@
         if (shouldAvoidProp(propsToAvoid, prop)) {
           continue
         }
-        if (Object.getOwnPropertyDescriptor(globalRef, prop)?.configurable === false) {
+        const globalRefPropertyDescriptor = Object.getOwnPropertyDescriptor(globalRef, prop);
+        if (globalRefPropertyDescriptor && globalRefPropertyDescriptor.configurable === false) {
           continue
         }
         const desc = {
