diff --git a/lib/lockdown.umd.js b/lib/lockdown.umd.js
index 2f32ec5014adc6006ad5148a759fbf48d46bb40c..6df14835df7ca594a775c7da639de7b657e229d7 100644
--- a/lib/lockdown.umd.js
+++ b/lib/lockdown.umd.js
@@ -8154,7 +8154,7 @@ function                tameFunctionConstructors() {
     FERAL_FUNCTION.prototype.constructor('return 1');
    }catch( ignore) {
     // Throws, no need to patch.
-    return harden({});
+    return {};
    }

   const newIntrinsics=  {};
