diff --git a/src/loadPolicy.js b/src/loadPolicy.js
index 0aa4f83d083bcf18db9cf69df45a4c52154af4fe..b481194d929070dc653b2a61295c5b9cfde5e129 100644
--- a/src/loadPolicy.js
+++ b/src/loadPolicy.js
@@ -104,10 +104,9 @@ async function loadPolicyAndApplyOverrides({
 
   const finalPolicy = mergePolicy(policy, policyOverride)
 
-  // TODO: Only write if merge results in changes.
-  // Would have to make a deep equal check on whole policy, which is a waste of time.
-  // mergePolicy() should be able to do it in one pass.
-  await fs.writeFile(policyPath, jsonStringifySortedPolicy(finalPolicy))
+  // Skip policy write step to prevent intermittent build failures
+  // The extension validates the policy in a separate step, we don't need it
+  // to be written to disk here.
 
   return finalPolicy
 }
