diff --git a/node_modules/lavamoat-core/lib/lockdown.umd.js b/node_modules/lavamoat-core/lib/lockdown.umd.js
index 2f32ec5..3336acc 100644
--- a/node_modules/lavamoat-core/lib/lockdown.umd.js
+++ b/node_modules/lavamoat-core/lib/lockdown.umd.js
@@ -8154,7 +8154,7 @@ function                tameFunctionConstructors() {
     FERAL_FUNCTION.prototype.constructor('return 1');
    }catch( ignore) {
     // Throws, no need to patch.
-    return harden({});
+    return {};
    }
 
   const newIntrinsics=  {};
