diff --git a/src/loadPolicy.js b/src/loadPolicy.js
index 0aa4f83d083bcf18db9cf69df45a4c52154af4fe..a830b026f4bf2ccc5be04dca9e9e081a771a8b43 100644
--- a/src/loadPolicy.js
+++ b/src/loadPolicy.js
@@ -104,11 +104,6 @@ async function loadPolicyAndApplyOverrides({
 
   const finalPolicy = mergePolicy(policy, policyOverride)
 
-  // TODO: Only write if merge results in changes.
-  // Would have to make a deep equal check on whole policy, which is a waste of time.
-  // mergePolicy() should be able to do it in one pass.
-  await fs.writeFile(policyPath, jsonStringifySortedPolicy(finalPolicy))
-
   return finalPolicy
 }
 
