diff --git a/src/inspectSource.js b/src/inspectSource.js
index 3d7f0b3ce2c1a022a5c031b38220587afd1f422a..3cac20564c798ed86e469d8c0af4556b8c703d09 100644
--- a/src/inspectSource.js
+++ b/src/inspectSource.js
@@ -90,10 +90,23 @@ function inspectGlobals(
           parents
         )
       )
+
+      // inspect globals passed to functions
+      if (parent.type === 'CallExpression') {
+        const narrowedUsages = inspectFunctionArgumentUsage(path, parent)
+        if (narrowedUsages) {
+          narrowedUsages.forEach(({ keys, use }) => {
+            maybeAddGlobalUsage([...keyPath, ...keys].join('.'), use)
+          })
+          return
+        }
+      }
+
       // if nested API lookup begins with a globalRef, drop it
       if (globalRefs.includes(keyPath[0])) {
         keyPath.shift()
       }
+
       // inspect for destructuring
       let destructuredPaths
       if (
@@ -158,6 +171,39 @@ function inspectGlobals(
     return { identifierUse, path, parent: parentOfMembershipChain }
   }
 
+  /**
+   * @param {import('@babel/traverse').NodePath<import('@babel/types').Identifier | import('@babel/types').ThisExpression>} path
+   * @param {import('@babel/types').CallExpression} callExpression
+   * @returns {{ keys: string[], use: GlobalPolicyValue }[] | null}
+   */
+  function inspectFunctionArgumentUsage(path, callExpression) {
+    const argIndex = callExpression.arguments.indexOf(path.node)
+    if (argIndex === -1 || callExpression.callee.type !== 'Identifier') {
+      return null
+    }
+    const functionPath = path.scope.getBinding(callExpression.callee.name)?.path
+    if (!functionPath?.isFunctionDeclaration()) {
+      return null
+    }
+    const param = functionPath?.node.params[argIndex]
+    // skip parameters that arent plain identifiers
+    if (param?.type !== 'Identifier') {
+      return null
+    }
+    const paramReferences = functionPath.scope.getBinding(param.name)?.referencePaths
+    if (!paramReferences || paramReferences.length === 0) {
+      return null
+    }
+    return paramReferences.map((ref) => {
+      const { path, identifierUse } = inspectIdentifierForDirectMembershipChain(
+        param.name,
+        /** @type {import('@babel/types').Identifier} */ (ref.node),
+        /** @type {import('./inspectPrimordialAssignments').MemberLikeExpression[]} */ (getParents(ref))
+      )
+      return { keys: path.slice(1), use: identifierUse }
+    })
+  }
+
   /**
    * @param {string} identifierPath
    * @param {import('@lavamoat/types').GlobalPolicyValue} identifierUse
