diff --git a/src/cli.js b/src/cli.js
index 32ea5fd66beaa73a780256e3f5b0d79008e97783..3d68a1590c3ee29e58c63c95ec7386989d440650 100755
--- a/src/cli.js
+++ b/src/cli.js
@@ -24,15 +24,16 @@ function parseArgs() {
       })
       yargsFlags(yargs, defaults)
     })
+    .parserConfiguration({
+      'populate--': true
+    })
     .help()
 
   const parsedArgs = argsParser.parse()
 
-  // patch process.argv so it matches the normal pattern
-  // e.g. [runtime path, entrypoint, ...args]
-  // we'll use the LavaMoat path as the runtime
-  // so we just remove the node path
-  process.argv.shift()
+  // Rebuild process.argv to pass only entry args
+  const entryArgs = parsedArgs['--'] || []
+  process.argv = [process.argv[0], parsedArgs.entryPath, ...entryArgs]
 
   return parsedArgs
 }
diff --git a/src/parseForPolicy.js b/src/parseForPolicy.js
index 7323cfedd5876fef5428c50313880e0cc68ef31b..7d5007c9b07eb3cdae047d008eab22f0f7025cfe 100644
--- a/src/parseForPolicy.js
+++ b/src/parseForPolicy.js
@@ -21,7 +21,8 @@ const { checkForResolutionOverride } = require('./resolutions')
 
 // file extension omitted can be omitted, eg https://npmfs.com/package/yargs/17.0.1/yargs
 const commonjsExtensions = ['', '.js', '.cjs']
-const resolutionOmittedExtensions = ['.js', '.json']
+// extensions we want to give to the resolver that it omits by default. .js is here duplicated for our own sanity mostly.
+const resolutionOmittedExtensions = ['.js', '.cjs', '.json']
 
 /**
  * Allow use of `node:` prefixed builtins.
