diff --git a/build/cjs-browser/luxon.js b/build/cjs-browser/luxon.js
index 9ab2b9fd75714368c9c71975ea280bf5d49cb237..14c2891068731bb8bd4ac834d22fb5525a5ed162 100644
--- a/build/cjs-browser/luxon.js
+++ b/build/cjs-browser/luxon.js
@@ -7373,7 +7373,7 @@ var DateTime = /*#__PURE__*/function () {
    */
   ;
 
-  _proto.toLocaleString = function toLocaleString(formatOpts, opts) {
+  Reflect.defineProperty(_proto, 'toLocaleString', { value: function toLocaleString(formatOpts, opts) {
     if (formatOpts === void 0) {
       formatOpts = DATE_SHORT;
     }
@@ -7383,7 +7383,7 @@ var DateTime = /*#__PURE__*/function () {
     }
 
     return this.isValid ? Formatter.create(this.loc.clone(opts), formatOpts).formatDateTime(this) : INVALID;
-  }
+  }})
   /**
    * Returns an array of format "parts", meaning individual tokens along with metadata. This is allows callers to post-process individual sections of the formatted output.
    * Defaults to the system's locale if no locale has been specified
