diff --git a/index.js b/index.js
index a4d360dc6c6343d321bf2bae46e743718a5cb480..d2be4a20f717d33c1268c860ca74287ac86597c4 100644
--- a/index.js
+++ b/index.js
@@ -54,7 +54,6 @@ function PluginError(plugin, message, options) {
       return this._messageWithDetails() + '\nStack:';
     }.bind(this);
 
-    Error.captureStackTrace(safety, arguments.callee || this.constructor);
     this.__safety = safety;
   }
   if (!this.plugin) {
