diff --git a/dist/squirrelly.cjs.js b/dist/squirrelly.cjs.js
<<<<<<<< HEAD:.yarn/patches/squirrelly-npm-9.0.0-3cf710c7bb.patch
index b87580be5da8a9a5af4cf60c8c5acf910c8f9ed2..e037b8a986d331f7075bbd448c80104de8a3453c 100644
========
index 4680ee747900853b9af01b480c749880dedb9824..95dce7623bc115508063a78c7de9ed7b6e4d3d82 100644
>>>>>>>> upstream/yarn-update-patches:.yarn/patches/squirrelly-npm-8.0.8-1d17420d8d.patch
--- a/dist/squirrelly.cjs.js
+++ b/dist/squirrelly.cjs.js
@@ -5,7 +5,7 @@ Object.defineProperty(exports, '__esModule', { value: true });
 // TODO: allow '-' to trim up until newline. Use [^\S\n\r] instead of \s
 // TODO: only include trimLeft polyfill if not in ES6
 /* END TYPES */
-var promiseImpl = new Function('return this')().Promise;
+var promiseImpl = globalThis.Promise;
 var asyncFunc = false;
 try {
     asyncFunc = new Function('return (async function(){}).constructor')();
