diff --git a/headers.js b/headers.js
index 56c3e9b26844d3e10a805cf850cdd09a172fa710..aa1a6c45d1623d9b94a5e065bd2872f0c4be4915 100644
--- a/headers.js
+++ b/headers.js
@@ -34,7 +34,7 @@ exports.decodePax = function decodePax (buf) {
   while (buf.length) {
     let i = 0
     while (i < buf.length && buf[i] !== 32) i++
-    const len = parseInt(buf.subarray(0, i).toString(), 10)
+    const len = parseInt(b4a.toString(buf.subarray(0, i)), 10)
     if (!len) return result
 
     const b = b4a.toString(buf.subarray(i + 1, len - 1))
@@ -304,7 +304,7 @@ function decodeOct (val, offset, length) {
     const end = clamp(indexOf(val, 32, offset, val.length), val.length, val.length)
     while (offset < end && val[offset] === 0) offset++
     if (end === offset) return 0
-    return parseInt(val.subarray(offset, end).toString(), 8)
+    return parseInt(b4a.toString(val.subarray(offset, end)), 8)
   }
 }
 
