diff --git a/node_modules/@eslint/eslintrc/lib/config-array-factory.js b/node_modules/@eslint/eslintrc/lib/config-array-factory.js
index 9553d1f..3ea3e55 100644
--- a/node_modules/@eslint/eslintrc/lib/config-array-factory.js
+++ b/node_modules/@eslint/eslintrc/lib/config-array-factory.js
@@ -40,7 +40,6 @@
 
 import debugOrig from "debug";
 import fs from "fs";
-import importFresh from "import-fresh";
 import { createRequire } from "module";
 import path from "path";
 import stripComments from "strip-json-comments";
@@ -231,7 +230,7 @@ function loadLegacyConfigFile(filePath) {
 function loadJSConfigFile(filePath) {
     debug(`Loading JS config file: ${filePath}`);
     try {
-        return importFresh(filePath);
+        return require(filePath);
     } catch (e) {
         debug(`Error reading JavaScript file: ${filePath}`);
         e.message = `Cannot read config file: ${filePath}\nError: ${e.message}`;
