diff --git a/node_modules/@formatjs/intl-utils/dist/index.js b/node_modules/@formatjs/intl-utils/dist/index.js
index cb44944..4ec2d32 100644
--- a/node_modules/@formatjs/intl-utils/dist/index.js
+++ b/node_modules/@formatjs/intl-utils/dist/index.js
@@ -25,7 +25,7 @@ exports.toRawFixed = polyfill_utils_1.toRawFixed;
 exports.toRawPrecision = polyfill_utils_1.toRawPrecision;
 exports.getMagnitude = polyfill_utils_1.getMagnitude;
 exports.repeat = polyfill_utils_1.repeat;
-exports.hasOwnProperty = polyfill_utils_1.hasOwnProperty;
+Object.defineProperty(exports, 'hasOwnProperty', { value: polyfill_utils_1.hasOwnProperty });
 exports.isWellFormedUnitIdentifier = polyfill_utils_1.isWellFormedUnitIdentifier;
 exports.defineProperty = polyfill_utils_1.defineProperty;
 var resolve_locale_1 = require("./resolve-locale");
diff --git a/node_modules/@formatjs/intl-utils/dist/polyfill-utils.js b/node_modules/@formatjs/intl-utils/dist/polyfill-utils.js
index 9306ef0..24859ac 100644
--- a/node_modules/@formatjs/intl-utils/dist/polyfill-utils.js
+++ b/node_modules/@formatjs/intl-utils/dist/polyfill-utils.js
@@ -5,7 +5,7 @@ var units_1 = require("./units");
 function hasOwnProperty(o, key) {
     return Object.prototype.hasOwnProperty.call(o, key);
 }
-exports.hasOwnProperty = hasOwnProperty;
+Object.defineProperty(exports, 'hasOwnProperty', { value: hasOwnProperty });
 /**
  * https://tc39.es/ecma262/#sec-toobject
  * @param arg