diff --git a/node_modules/@lavamoat/lavapack/src/pack.js b/node_modules/@lavamoat/lavapack/src/pack.js
index eb41a0a..3f891ea 100644
--- a/node_modules/@lavamoat/lavapack/src/pack.js
+++ b/node_modules/@lavamoat/lavapack/src/pack.js
@@ -203,7 +203,9 @@ function createPacker({
     const jsonSerializeableData = {
       // id,
       package: packageName,
-      file,
+      // Omit this absolute filename from bundle so that builds are reproducible between environments
+      // TODO: update lavapack with an option to omit this, and/or make this filepath relative to the current working directory
+      // file,
       // deps,
       // source: sourceMeta.code
     }
diff --git a/node_modules/@lavamoat/lavapack/src/runtime.js b/node_modules/@lavamoat/lavapack/src/runtime.js
index 58f76f3..014fced 100644
--- a/node_modules/@lavamoat/lavapack/src/runtime.js
+++ b/node_modules/@lavamoat/lavapack/src/runtime.js
@@ -9982,6 +9982,13 @@ markVirtualizedNativeFunction)
           // eslint-disable-next-line @endo/no-polymorphic-call
           console.log(`Removing ${subPath}`);
         }
+        // are not accessible in strict-mode, so check to avoid the exception
+        if ((prop !== 'arguments' && prop !== 'caller' && prop !== 'callee')) {
+          // to avoid exception of trying to delete non deletable props set by Securely
+          if (obj[prop] === undefined) {
+            continue;
+          }
+        }
         try {
           delete obj[prop];
         } catch (err) {
