diff --git a/node_modules/@metamask/bip39/src/_wordlists.js b/node_modules/@metamask/bip39/src/_wordlists.js
index 4cf7f47..19f3157 100644
--- a/node_modules/@metamask/bip39/src/_wordlists.js
+++ b/node_modules/@metamask/bip39/src/_wordlists.js
@@ -6,53 +6,6 @@ Object.defineProperty(exports, "__esModule", { value: true });
 const wordlists = {};
 exports.wordlists = wordlists;
 let _default;
-exports._default = _default;
-try {
-    exports._default = _default = require('./wordlists/czech.json');
-    wordlists.czech = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/chinese_simplified.json');
-    wordlists.chinese_simplified = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/chinese_traditional.json');
-    wordlists.chinese_traditional = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/korean.json');
-    wordlists.korean = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/french.json');
-    wordlists.french = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/italian.json');
-    wordlists.italian = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/spanish.json');
-    wordlists.spanish = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/japanese.json');
-    wordlists.japanese = _default;
-    wordlists.JA = _default;
-}
-catch (err) { }
-try {
-    exports._default = _default = require('./wordlists/portuguese.json');
-    wordlists.portuguese = _default;
-}
-catch (err) { }
 try {
     exports._default = _default = require('./wordlists/english.json');
     wordlists.english = _default;
