diff --git a/node_modules/@reduxjs/toolkit/dist/redux-toolkit.cjs.development.js b/node_modules/@reduxjs/toolkit/dist/redux-toolkit.cjs.development.js
index 96ead94..5cd00c8 100644
--- a/node_modules/@reduxjs/toolkit/dist/redux-toolkit.cjs.development.js
+++ b/node_modules/@reduxjs/toolkit/dist/redux-toolkit.cjs.development.js
@@ -204,7 +204,7 @@ function (_Array) {
 
   var _proto = MiddlewareArray.prototype;
 
-  _proto.concat = function concat() {
+  Object.defineProperty(_proto, 'concat', { value: function concat() {
     var _Array$prototype$conc;
 
     for (var _len = arguments.length, arr = new Array(_len), _key = 0; _key < _len; _key++) {
@@ -212,7 +212,7 @@ function (_Array) {
     }
 
     return _construct(MiddlewareArray, (_Array$prototype$conc = _Array.prototype.concat).call.apply(_Array$prototype$conc, [this].concat(arr)));
-  };
+  } })
 
   _proto.prepend = function prepend() {
     for (var _len2 = arguments.length, arr = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
diff --git a/node_modules/@reduxjs/toolkit/dist/redux-toolkit.esm.js b/node_modules/@reduxjs/toolkit/dist/redux-toolkit.esm.js
index 859834f..4e14721 100644
--- a/node_modules/@reduxjs/toolkit/dist/redux-toolkit.esm.js
+++ b/node_modules/@reduxjs/toolkit/dist/redux-toolkit.esm.js
@@ -202,7 +202,7 @@ function (_Array) {
 
   var _proto = MiddlewareArray.prototype;
 
-  _proto.concat = function concat() {
+  Object.defineProperty(_proto, 'concat', { value: function concat() {
     var _Array$prototype$conc;
 
     for (var _len = arguments.length, arr = new Array(_len), _key = 0; _key < _len; _key++) {
@@ -210,7 +210,7 @@ function (_Array) {
     }
 
     return _construct(MiddlewareArray, (_Array$prototype$conc = _Array.prototype.concat).call.apply(_Array$prototype$conc, [this].concat(arr)));
-  };
+  } })
 
   _proto.prepend = function prepend() {
     for (var _len2 = arguments.length, arr = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
diff --git a/node_modules/@reduxjs/toolkit/dist/redux-toolkit.umd.js b/node_modules/@reduxjs/toolkit/dist/redux-toolkit.umd.js
index 185aa69..acb4c2a 100644
--- a/node_modules/@reduxjs/toolkit/dist/redux-toolkit.umd.js
+++ b/node_modules/@reduxjs/toolkit/dist/redux-toolkit.umd.js
@@ -1017,7 +1017,7 @@ function (_Array) {
 
   var _proto = MiddlewareArray.prototype;
 
-  _proto.concat = function concat() {
+  Object.defineProperty(_proto, 'concat', { value: function concat() {
     var _Array$prototype$conc;
 
     for (var _len = arguments.length, arr = new Array(_len), _key = 0; _key < _len; _key++) {
@@ -1025,7 +1025,7 @@ function (_Array) {
     }
 
     return _construct(MiddlewareArray, (_Array$prototype$conc = _Array.prototype.concat).call.apply(_Array$prototype$conc, [this].concat(arr)));
-  };
+  } })
 
   _proto.prepend = function prepend() {
     for (var _len2 = arguments.length, arr = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
