diff --git a/node_modules/async/internal/iterator.js b/node_modules/async/internal/iterator.js
index 3d32942..886beb5 100644
--- a/node_modules/async/internal/iterator.js
+++ b/node_modules/async/internal/iterator.js
@@ -43,6 +43,9 @@ function createObjectIterator(obj) {
     var len = okeys.length;
     return function next() {
         var key = okeys[++i];
+        if (key === '__proto__') {
+            return next();
+        }
         return i < len ? { value: obj[key], key: key } : null;
     };
 }
