diff --git a/node_modules/common-shakeify/index.js b/node_modules/common-shakeify/index.js
index 1e86067..c4fcee3 100644
--- a/node_modules/common-shakeify/index.js
+++ b/node_modules/common-shakeify/index.js
@@ -1,5 +1,13 @@
 'use strict'
-const relative = require('path').relative
+const _relative = require('path').relative
+const relative = (...args) => {
+  // console.warn('relative', args)
+  try {
+    return _relative(...args)
+  } catch (err) {
+    return `relative-failed:args=${args}`
+  }
+}
 const Analyzer = require('@goto-bus-stop/common-shake').Analyzer
 const transformAst = require('transform-ast')
 const wrapComment = require('wrap-comment')
@@ -82,7 +90,7 @@ function createStream (opts) {
     let ast
     const string = transformAst(source, {
       locations: true,
-      ecmaVersion: 9,
+      // ecmaVersion: 9,
       inputFilename: row.file
     }, (node) => {
       if (node.type === 'Program') ast = node
