diff --git a/node_modules/eth-query/index.js b/node_modules/eth-query/index.js
index 13e9f3c..d714bb7 100644
--- a/node_modules/eth-query/index.js
+++ b/node_modules/eth-query/index.js
@@ -1,5 +1,6 @@
 const extend = require('xtend')
 const createRandomId = require('json-rpc-random-id')()
+const debug = require('debug')('eth-query');
 
 module.exports = EthQuery
 
@@ -63,7 +64,10 @@ EthQuery.prototype.submitHashrate =                      generateFnFor('eth_subm
 
 EthQuery.prototype.sendAsync = function(opts, cb){
   const self = this
-  self.currentProvider.sendAsync(createPayload(opts), function(err, response){
+  const payload = createPayload(opts)
+  debug('making request %o', payload)
+  self.currentProvider.sendAsync(payload, function(err, response){
+    debug('got err = %o, response = %o', err, response)
     if (!err && response.error) err = new Error('EthQuery - RPC Error - '+response.error.message)
     if (err) return cb(err)
     cb(null, response.result)
