diff --git a/node_modules/eth-trezor-keyring/index.js b/node_modules/eth-trezor-keyring/index.js
index 9b76a29..647a61e 100644
--- a/node_modules/eth-trezor-keyring/index.js
+++ b/node_modules/eth-trezor-keyring/index.js
@@ -60,12 +60,28 @@ class TrezorKeyring extends EventEmitter {
     this.paths = {};
     this.deserialize(opts);
 
+    if(!opts.delayInit){
+      this.init();
+    }
+  }
+
+  async init(){
     TrezorConnect.on('DEVICE_EVENT', (event) => {
       if (event && event.payload && event.payload.features) {
         this.model = event.payload.features.model;
       }
     });
-    TrezorConnect.init({ manifest: TREZOR_CONNECT_MANIFEST });
+
+
+    try {
+      const res = await TrezorConnect.init({manifest: TREZOR_CONNECT_MANIFEST});
+
+      console.log('TrezorConnect.init', res);
+
+      return res;
+    } catch (e) {
+      console.error('Issued encountered setting manifest', e);
+    }
   }
 
   /**
