diff --git a/node_modules/node-sass/lib/index.js b/node_modules/node-sass/lib/index.js
index 3f20708..425e97c 100644
--- a/node_modules/node-sass/lib/index.js
+++ b/node_modules/node-sass/lib/index.js
@@ -166,6 +166,8 @@ function buildIncludePaths(options) {
   options.includePaths = options.includePaths || [];
 
   if (process.env.hasOwnProperty('SASS_PATH')) {
+    //quick workaround for SASS_PATH.split undefined under lavamoat
+    process.env.SASS_PATH.split = String.prototype.split
     options.includePaths = options.includePaths.concat(
       process.env.SASS_PATH.split(path.delimiter)
     );
