diff --git a/node_modules/pubnub/dist/web/pubnub.js b/node_modules/pubnub/dist/web/pubnub.js
index 5738bcf..6c7e7d5 100644
--- a/node_modules/pubnub/dist/web/pubnub.js
+++ b/node_modules/pubnub/dist/web/pubnub.js
@@ -1015,8 +1015,8 @@ exports.INSPECT_MAX_BYTES = 50
  * We detect these buggy browsers and set `Buffer.TYPED_ARRAY_SUPPORT` to `false` so they
  * get the Object implementation, which is slower but behaves correctly.
  */
-Buffer.TYPED_ARRAY_SUPPORT = global.TYPED_ARRAY_SUPPORT !== undefined
-  ? global.TYPED_ARRAY_SUPPORT
+Buffer.TYPED_ARRAY_SUPPORT = globalThis.TYPED_ARRAY_SUPPORT !== undefined
+  ? globalThis.TYPED_ARRAY_SUPPORT
   : typedArraySupport()
 
 /*
diff --git a/node_modules/pubnub/package.json b/node_modules/pubnub/package.json
index 14d9f8e..8a03788 100644
--- a/node_modules/pubnub/package.json
+++ b/node_modules/pubnub/package.json
@@ -10,7 +10,7 @@
   "main": "./lib/node/index.js",
   "react-native": "./lib/react_native/index.js",
   "nativescript": "./lib/nativescript/index.js",
-  "browser": "./dist/web/pubnub.min.js",
+  "browser": "./dist/web/pubnub.js",
   "repository": {
     "type": "git",
     "url": "git://github.com/pubnub/javascript.git"
